<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>2 Analysis | Language-like efficiency and structure in house finch song</title>
  <meta name="description" content="2 Analysis | Language-like efficiency and structure in house finch song" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="2 Analysis | Language-like efficiency and structure in house finch song" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="https://masonyoungblood.github.io/linguistic_efficiency/featured.jpg" />
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="2 Analysis | Language-like efficiency and structure in house finch song" />
  
  
  <meta name="twitter:image" content="https://masonyoungblood.github.io/linguistic_efficiency/featured.jpg" />




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="introduction.html"/>
<link rel="next" href="discussion.html"/>
<style type="text/css">
p.abstract{
  text-align: center;
  font-weight: bold;
}
div.abstract{
  margin: auto;
  width: 90%;
}
</style>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/codefolding-lua-1.1/codefolding-lua.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/tabwid-1.1.3/tabwid.css" rel="stylesheet" />
<script src="libs/tabwid-1.1.3/tabwid.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="links.html"><a href="links.html"><i class="fa fa-check"></i>Links</a></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="introduction.html"><a href="introduction.html#efficiency-complexity"><i class="fa fa-check"></i><b>1.1</b> Efficiency &amp; Complexity</a></li>
<li class="chapter" data-level="1.2" data-path="introduction.html"><a href="introduction.html#linguistic-laws-structure"><i class="fa fa-check"></i><b>1.2</b> Linguistic Laws &amp; Structure</a></li>
<li class="chapter" data-level="1.3" data-path="introduction.html"><a href="introduction.html#granularity"><i class="fa fa-check"></i><b>1.3</b> Granularity</a></li>
<li class="chapter" data-level="1.4" data-path="introduction.html"><a href="introduction.html#aim-model"><i class="fa fa-check"></i><b>1.4</b> Aim &amp; Model</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="analysis.html"><a href="analysis.html"><i class="fa fa-check"></i><b>2</b> Analysis</a>
<ul>
<li class="chapter" data-level="2.1" data-path="analysis.html"><a href="analysis.html#data"><i class="fa fa-check"></i><b>2.1</b> Data</a></li>
<li class="chapter" data-level="2.2" data-path="analysis.html"><a href="analysis.html#clustering"><i class="fa fa-check"></i><b>2.2</b> Clustering</a></li>
<li class="chapter" data-level="2.3" data-path="analysis.html"><a href="analysis.html#zipfs-rank-frequency-law"><i class="fa fa-check"></i><b>2.3</b> Zipf’s Rank-Frequency Law</a></li>
<li class="chapter" data-level="2.4" data-path="analysis.html"><a href="analysis.html#zipfs-law-of-abbreviation"><i class="fa fa-check"></i><b>2.4</b> Zipf’s Law of Abbreviation</a></li>
<li class="chapter" data-level="2.5" data-path="analysis.html"><a href="analysis.html#menzeraths-law"><i class="fa fa-check"></i><b>2.5</b> Menzerath’s Law</a></li>
<li class="chapter" data-level="2.6" data-path="analysis.html"><a href="analysis.html#small-worldness-index"><i class="fa fa-check"></i><b>2.6</b> Small-Worldness Index</a></li>
<li class="chapter" data-level="2.7" data-path="analysis.html"><a href="analysis.html#mutual-information"><i class="fa fa-check"></i><b>2.7</b> Mutual Information</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="discussion.html"><a href="discussion.html"><i class="fa fa-check"></i><b>3</b> Discussion</a></li>
<li class="chapter" data-level="" data-path="acknowledgments.html"><a href="acknowledgments.html"><i class="fa fa-check"></i>Acknowledgments</a></li>
<li class="chapter" data-level="" data-path="data-code-availability.html"><a href="data-code-availability.html"><i class="fa fa-check"></i>Data &amp; Code Availability</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Language-like efficiency and structure in house finch song</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="analysis" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">2</span> Analysis<a href="analysis.html#analysis" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Unless otherwise stated, all models were fit in STAN using the brms package in R <span class="citation">(<a href="#ref-burkner17">81</a>)</span>, with 20,000 iterations across four MCMC chains. Model comparison was conducted using the widely-applicable information criterion (WAIC), a more robust alternative to AIC <span class="citation">(<a href="#ref-watanabe21">82</a>)</span>, and a Bayesian form of R<sup>2</sup> <span class="citation">(<a href="#ref-gelman_etal19">83</a>)</span>. Prior specifications, model diagnostics, and full model output for all analyses can be found in the <a href="https://masonyoungblood.github.io/linguistic_efficiency/supplement/supplement.html">Supplementary Information</a>. <!--I use the diagnostic heuristics recommended by @gelman_etal14: effective sample size $> 100$, and $1 < \hat{R} < 1.1$.--></p>
<div id="data" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> Data<a href="analysis.html#data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The recordings used in this study (2,724 songs from 331 individuals) were collected in 1975 <span class="citation">(<a href="#ref-Mundinger1975">74</a>)</span>, 2012 <span class="citation">(<a href="#ref-Ju2019">75</a>)</span>, and 2019 <span class="citation">(<a href="#ref-youngbloodContentBiasCultural2022">9</a>)</span> in the New York metropolitan area, and analyzed by <span class="citation">(<a href="#ref-youngbloodContentBiasCultural2022">9</a>)</span> using Luscinia (<a href="https://rflachlan.github.io/Luscinia/" class="uri">https://rflachlan.github.io/Luscinia/</a>) (full recording and analysis details, and an example of an analyzed song, are in <span class="citation">(<a href="#ref-youngbloodContentBiasCultural2022">9</a>)</span> and the <a href="https://masonyoungblood.github.io/linguistic_efficiency/supplement/supplement.html">Supplementary Information</a>). The main analysis was conducted using recordings from all three years, but the patterns are qualitatively the same when each year is analyzed separately (see <a href="https://masonyoungblood.github.io/linguistic_efficiency/supplement/supplement.html">Supplementary Information</a>).</p>
<!-- The data used in this study comes directly from the repository of @youngbloodContentBiasCultural2022 (https://github.com/masonyoungblood/TransmissionBias). -->
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="analysis.html#cb1-1" tabindex="-1"></a><span class="co">#load data from repository for Youngblood &amp; Lahti (2022)</span></span>
<span id="cb1-2"><a href="analysis.html#cb1-2" tabindex="-1"></a><span class="fu">load</span>(<span class="fu">url</span>(<span class="st">&quot;https://github.com/masonyoungblood/TransmissionBias/raw/master/example/data.RData&quot;</span>))</span></code></pre></div>
</details>
</div>
<div id="clustering" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> Clustering<a href="analysis.html#clustering" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Clustering was conducted using the log-transformed mean frequency traces (mean frequency in each 1 ms bin) for each syllable in every song <span class="citation">(<a href="#ref-Lachlan2018">84</a>)</span>. First, the normalized distances between all of the syllables were calculated via dynamic time warping with a window size of 10 (10% of the average signal length) using the dtwclust package in R <span class="citation">(<a href="#ref-SardaEspinosa2019">85</a>)</span>. A window size of 10% of the signal length is commonly used in speech processing research and seems to be a practical upper limit for many applications <span class="citation">(<a href="#ref-Ratanamahatana2005">86</a>)</span>. Infinite distances (0.19% of cases) caused by comparisons of syllables with extreme signal length differences were assigned the maximum observed distance value. Next, hierarchical clustering and dynamic tree cut were used to cluster the syllables into types <span class="citation">(<a href="#ref-Roginek2018">87</a>)</span>. Hierarchical clustering was conducted with the UPGMA method implemented in the fastcluster package in R <span class="citation">(<a href="#ref-JSSv053i09">88</a>)</span>, and dynamic tree cut was conducted with the dynamicTreeCut package in R <span class="citation">(<a href="#ref-langfelder_etal08">89</a>)</span>.</p>
<p>For dynamic tree cut, I ran the hybrid algorithm with a minimum cluster size of 1 to maximize the representation of rare syllable types, and used the deep split parameter (<span class="math inline">\(DS\)</span>) to determine the granularity of clustering (details of <span class="math inline">\(DS\)</span> are in the <a href="https://masonyoungblood.github.io/linguistic_efficiency/supplement/supplement.html">Supplementary Information</a>). I restricted this analysis to <span class="math inline">\(DS = \{2, 3, 4\}\)</span> because values of <span class="math inline">\(DS = \{0, 1\}\)</span> lead to extremely unrealistic underestimates of the number of syllables types. <span class="math inline">\(DS = 2\)</span> leads to over-lumping of syllables into types (<span class="math inline">\(n = 114\)</span>; low granularity), <span class="math inline">\(DS = 3\)</span> leads to a typical syllable classification (<span class="math inline">\(n = 596\)</span>) <span class="citation">(<a href="#ref-youngbloodContentBiasCultural2022">9</a>,<a href="#ref-Ju2019">75</a>,<a href="#ref-Roginek2018">87</a>)</span>, and <span class="math inline">\(DS = 4\)</span> lead to over-splitting (<span class="math inline">\(n = 1646\)</span>; high granularity). The dendrogram and syllable classifications can be seen in Figure <a href="analysis.html#fig:figure-clustering">1</a>.</p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="analysis.html#cb2-1" tabindex="-1"></a><span class="co">#load library</span></span>
<span id="cb2-2"><a href="analysis.html#cb2-2" tabindex="-1"></a><span class="fu">library</span>(dtwclust)</span>
<span id="cb2-3"><a href="analysis.html#cb2-3" tabindex="-1"></a></span>
<span id="cb2-4"><a href="analysis.html#cb2-4" tabindex="-1"></a><span class="co">#log transform each syllable individually</span></span>
<span id="cb2-5"><a href="analysis.html#cb2-5" tabindex="-1"></a>data<span class="sc">$</span>meanfreq <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(data), <span class="cf">function</span>(x){<span class="fu">log</span>(data<span class="sc">$</span>meanfreq[[x]])})</span>
<span id="cb2-6"><a href="analysis.html#cb2-6" tabindex="-1"></a></span>
<span id="cb2-7"><a href="analysis.html#cb2-7" tabindex="-1"></a><span class="co">#z-score normalization across entire dataset</span></span>
<span id="cb2-8"><a href="analysis.html#cb2-8" tabindex="-1"></a>m <span class="ot">&lt;-</span> <span class="fu">mean</span>(<span class="fu">unlist</span>(data<span class="sc">$</span>meanfreq))</span>
<span id="cb2-9"><a href="analysis.html#cb2-9" tabindex="-1"></a>sd <span class="ot">&lt;-</span> <span class="fu">sd</span>(<span class="fu">unlist</span>(data<span class="sc">$</span>meanfreq))</span>
<span id="cb2-10"><a href="analysis.html#cb2-10" tabindex="-1"></a>data<span class="sc">$</span>meanfreq <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(data), <span class="cf">function</span>(x){(data<span class="sc">$</span>meanfreq[[x]]<span class="sc">-</span>m)<span class="sc">/</span>sd})</span>
<span id="cb2-11"><a href="analysis.html#cb2-11" tabindex="-1"></a></span>
<span id="cb2-12"><a href="analysis.html#cb2-12" tabindex="-1"></a><span class="co">#set window size</span></span>
<span id="cb2-13"><a href="analysis.html#cb2-13" tabindex="-1"></a>window <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fu">mean</span>(<span class="fu">lengths</span>(data<span class="sc">$</span>meanfreq))<span class="sc">*</span><span class="fl">0.10</span>) <span class="co">#10</span></span>
<span id="cb2-14"><a href="analysis.html#cb2-14" tabindex="-1"></a></span>
<span id="cb2-15"><a href="analysis.html#cb2-15" tabindex="-1"></a><span class="co">#calculate distances</span></span>
<span id="cb2-16"><a href="analysis.html#cb2-16" tabindex="-1"></a>distances <span class="ot">&lt;-</span> proxy<span class="sc">::</span><span class="fu">dist</span>(data<span class="sc">$</span>meanfreq, <span class="at">method =</span> <span class="st">&quot;dtw_basic&quot;</span>, <span class="at">window.size =</span> window, <span class="at">normalize =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-17"><a href="analysis.html#cb2-17" tabindex="-1"></a></span>
<span id="cb2-18"><a href="analysis.html#cb2-18" tabindex="-1"></a><span class="co">#strip everything from distances</span></span>
<span id="cb2-19"><a href="analysis.html#cb2-19" tabindex="-1"></a><span class="fu">attr</span>(distances, <span class="st">&quot;dimnames&quot;</span>) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb2-20"><a href="analysis.html#cb2-20" tabindex="-1"></a><span class="fu">attr</span>(distances, <span class="st">&quot;method&quot;</span>) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb2-21"><a href="analysis.html#cb2-21" tabindex="-1"></a><span class="fu">attr</span>(distances, <span class="st">&quot;call&quot;</span>) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb2-22"><a href="analysis.html#cb2-22" tabindex="-1"></a><span class="fu">class</span>(distances) <span class="ot">&lt;-</span> <span class="st">&quot;matrix&quot;</span></span>
<span id="cb2-23"><a href="analysis.html#cb2-23" tabindex="-1"></a></span>
<span id="cb2-24"><a href="analysis.html#cb2-24" tabindex="-1"></a><span class="co">#replace infinite distances with the maximum value</span></span>
<span id="cb2-25"><a href="analysis.html#cb2-25" tabindex="-1"></a>distances[<span class="fu">which</span>(distances <span class="sc">==</span> <span class="cn">Inf</span>)] <span class="ot">&lt;-</span> <span class="fu">max</span>(distances[<span class="sc">-</span><span class="fu">which</span>(distances <span class="sc">==</span> <span class="cn">Inf</span>)])</span>
<span id="cb2-26"><a href="analysis.html#cb2-26" tabindex="-1"></a></span>
<span id="cb2-27"><a href="analysis.html#cb2-27" tabindex="-1"></a><span class="co">#convert to format compatible with hclust</span></span>
<span id="cb2-28"><a href="analysis.html#cb2-28" tabindex="-1"></a>dtw_dist <span class="ot">&lt;-</span> stats<span class="sc">::</span><span class="fu">as.dist</span>(distances)</span>
<span id="cb2-29"><a href="analysis.html#cb2-29" tabindex="-1"></a></span>
<span id="cb2-30"><a href="analysis.html#cb2-30" tabindex="-1"></a><span class="co">#run clustering</span></span>
<span id="cb2-31"><a href="analysis.html#cb2-31" tabindex="-1"></a>clustering <span class="ot">&lt;-</span> fastcluster<span class="sc">::</span><span class="fu">hclust</span>(dtw_dist, <span class="at">method =</span> <span class="st">&quot;average&quot;</span>)</span>
<span id="cb2-32"><a href="analysis.html#cb2-32" tabindex="-1"></a></span>
<span id="cb2-33"><a href="analysis.html#cb2-33" tabindex="-1"></a><span class="co">#save clustering</span></span>
<span id="cb2-34"><a href="analysis.html#cb2-34" tabindex="-1"></a><span class="fu">save</span>(clustering, <span class="at">file =</span> <span class="st">&quot;data_models/clustering.RData&quot;</span>)</span>
<span id="cb2-35"><a href="analysis.html#cb2-35" tabindex="-1"></a></span>
<span id="cb2-36"><a href="analysis.html#cb2-36" tabindex="-1"></a><span class="co">#run hybrid tree cut</span></span>
<span id="cb2-37"><a href="analysis.html#cb2-37" tabindex="-1"></a>hybrid_cut <span class="ot">&lt;-</span> parallel<span class="sc">::</span><span class="fu">mclapply</span>(<span class="dv">0</span><span class="sc">:</span><span class="dv">4</span>, <span class="cf">function</span>(x){dynamicTreeCut<span class="sc">::</span><span class="fu">cutreeDynamic</span>(<span class="at">dendro =</span> clustering, <span class="at">distM =</span> distances, <span class="at">method =</span> <span class="st">&quot;hybrid&quot;</span>, <span class="at">minClusterSize =</span> <span class="dv">1</span>, <span class="at">deepSplit =</span> x)}, <span class="at">mc.cores =</span> <span class="dv">5</span>)</span>
<span id="cb2-38"><a href="analysis.html#cb2-38" tabindex="-1"></a></span>
<span id="cb2-39"><a href="analysis.html#cb2-39" tabindex="-1"></a><span class="co">#save hybrid tree cut</span></span>
<span id="cb2-40"><a href="analysis.html#cb2-40" tabindex="-1"></a><span class="fu">save</span>(hybrid_cut, <span class="at">file =</span> <span class="st">&quot;data_models/hybrid_cut.RData&quot;</span>)</span></code></pre></div>
</details>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:figure-clustering"></span>
<img src="plots/clustering.png" alt="The results of hierarchical clustering. The colored bars below the dendrogram correspond to the categories assigned to each syllable when deep split is 2 (over-lumping), 3 (baseline), and 4 (over-splitting)." width="100%" />
<p class="caption">
Figure 1: The results of hierarchical clustering. The colored bars below the dendrogram correspond to the categories assigned to each syllable when deep split is 2 (over-lumping), 3 (baseline), and 4 (over-splitting).
</p>
</div>
<p>Once syllable types were identified by hierarchical clustering, I followed <span class="citation">(<a href="#ref-Ju2019">75</a>)</span> and <span class="citation">(<a href="#ref-youngbloodContentBiasCultural2022">9</a>)</span> in calculating the following syllable-level acoustic features for further analysis: average frequency (Hz), minimum frequency (Hz), maximum frequency (Hz), bandwidth (Hz), duration (ms), concavity (changes in the sign of the slope of the mean frequency trace/ms) and excursion (cumulative absolute change in Hz/ms). Concavity and excursion are both indicators of syllable complexity <span class="citation">(<a href="#ref-Ju2019">75</a>,<a href="#ref-podosFinescaleBroadlyApplicable2016">90</a>)</span>. Concavity was calculated after smoothing the mean frequency trace using a polynomial spline with a smoothing parameter of 5 <span class="citation">(<a href="#ref-youngbloodContentBiasCultural2022">9</a>)</span>.</p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="analysis.html#cb3-1" tabindex="-1"></a><span class="co">#load data from clustering</span></span>
<span id="cb3-2"><a href="analysis.html#cb3-2" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;data_models/clustering.RData&quot;</span>)</span>
<span id="cb3-3"><a href="analysis.html#cb3-3" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;data_models/hybrid_cut.RData&quot;</span>)</span>
<span id="cb3-4"><a href="analysis.html#cb3-4" tabindex="-1"></a></span>
<span id="cb3-5"><a href="analysis.html#cb3-5" tabindex="-1"></a><span class="co">#calculate concavity</span></span>
<span id="cb3-6"><a href="analysis.html#cb3-6" tabindex="-1"></a>inds <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(data)</span>
<span id="cb3-7"><a href="analysis.html#cb3-7" tabindex="-1"></a>inds <span class="ot">&lt;-</span> inds[<span class="sc">-</span><span class="fu">which</span>(<span class="fu">lengths</span>(data<span class="sc">$</span>meanfreq) <span class="sc">&lt;</span> <span class="dv">6</span>)] <span class="co">#remove things that are too short to analyze but have a concavity of 0</span></span>
<span id="cb3-8"><a href="analysis.html#cb3-8" tabindex="-1"></a>concavity <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="dv">0</span>, <span class="fu">nrow</span>(data))</span>
<span id="cb3-9"><a href="analysis.html#cb3-9" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> inds){</span>
<span id="cb3-10"><a href="analysis.html#cb3-10" tabindex="-1"></a>  temp <span class="ot">&lt;-</span> pspline<span class="sc">::</span><span class="fu">sm.spline</span>(data<span class="sc">$</span>meanfreq[[i]], <span class="at">spar =</span> <span class="dv">5</span>)</span>
<span id="cb3-11"><a href="analysis.html#cb3-11" tabindex="-1"></a>  temp <span class="ot">&lt;-</span> <span class="fu">diff</span>(<span class="fu">c</span>(temp<span class="sc">$</span>ysmth)) <span class="co">#extract first derivative (slopes)</span></span>
<span id="cb3-12"><a href="analysis.html#cb3-12" tabindex="-1"></a>  concav <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb3-13"><a href="analysis.html#cb3-13" tabindex="-1"></a>  <span class="cf">for</span>(j <span class="cf">in</span> <span class="dv">2</span><span class="sc">:</span><span class="fu">length</span>(temp)){</span>
<span id="cb3-14"><a href="analysis.html#cb3-14" tabindex="-1"></a>    <span class="cf">if</span>((temp[[j]] <span class="sc">&lt;</span> <span class="dv">0</span> <span class="sc">&amp;</span> temp[[j<span class="dv">-1</span>]] <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">|</span> (temp[[j]] <span class="sc">&gt;</span> <span class="dv">0</span> <span class="sc">&amp;</span> temp[[j<span class="dv">-1</span>]] <span class="sc">&lt;</span> <span class="dv">0</span>)){</span>
<span id="cb3-15"><a href="analysis.html#cb3-15" tabindex="-1"></a>      concavity[i] <span class="ot">&lt;-</span> concavity[i] <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb3-16"><a href="analysis.html#cb3-16" tabindex="-1"></a>    }</span>
<span id="cb3-17"><a href="analysis.html#cb3-17" tabindex="-1"></a>  }</span>
<span id="cb3-18"><a href="analysis.html#cb3-18" tabindex="-1"></a>  concavity[i] <span class="ot">&lt;-</span> concavity[i]<span class="sc">/</span><span class="fu">length</span>(temp)</span>
<span id="cb3-19"><a href="analysis.html#cb3-19" tabindex="-1"></a>  <span class="co">#concavity[i] &lt;- concavity[i]</span></span>
<span id="cb3-20"><a href="analysis.html#cb3-20" tabindex="-1"></a>}</span>
<span id="cb3-21"><a href="analysis.html#cb3-21" tabindex="-1"></a></span>
<span id="cb3-22"><a href="analysis.html#cb3-22" tabindex="-1"></a><span class="co">#calculate excursion</span></span>
<span id="cb3-23"><a href="analysis.html#cb3-23" tabindex="-1"></a>excursion <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="dv">0</span>, <span class="fu">nrow</span>(data))</span>
<span id="cb3-24"><a href="analysis.html#cb3-24" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> inds){</span>
<span id="cb3-25"><a href="analysis.html#cb3-25" tabindex="-1"></a>  temp <span class="ot">&lt;-</span> data<span class="sc">$</span>meanfreq[[i]]</span>
<span id="cb3-26"><a href="analysis.html#cb3-26" tabindex="-1"></a>  excursion[i] <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">sapply</span>(<span class="dv">2</span><span class="sc">:</span><span class="fu">length</span>(temp), <span class="cf">function</span>(x){<span class="fu">abs</span>(temp[x]<span class="sc">-</span>temp[x<span class="dv">-1</span>])}))<span class="sc">/</span><span class="fu">length</span>(temp)</span>
<span id="cb3-27"><a href="analysis.html#cb3-27" tabindex="-1"></a>}</span>
<span id="cb3-28"><a href="analysis.html#cb3-28" tabindex="-1"></a></span>
<span id="cb3-29"><a href="analysis.html#cb3-29" tabindex="-1"></a><span class="co">#add to data table</span></span>
<span id="cb3-30"><a href="analysis.html#cb3-30" tabindex="-1"></a>data<span class="sc">$</span>concavity <span class="ot">&lt;-</span> concavity</span>
<span id="cb3-31"><a href="analysis.html#cb3-31" tabindex="-1"></a>data<span class="sc">$</span>excursion <span class="ot">&lt;-</span> excursion</span>
<span id="cb3-32"><a href="analysis.html#cb3-32" tabindex="-1"></a></span>
<span id="cb3-33"><a href="analysis.html#cb3-33" tabindex="-1"></a><span class="co">#calculate other measures from mean frequency trace</span></span>
<span id="cb3-34"><a href="analysis.html#cb3-34" tabindex="-1"></a>data<span class="sc">$</span>average <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(data), <span class="cf">function</span>(x){<span class="fu">mean</span>(data<span class="sc">$</span>meanfreq[[x]])})</span>
<span id="cb3-35"><a href="analysis.html#cb3-35" tabindex="-1"></a>data<span class="sc">$</span>min <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(data), <span class="cf">function</span>(x){<span class="fu">min</span>(data<span class="sc">$</span>meanfreq[[x]])})</span>
<span id="cb3-36"><a href="analysis.html#cb3-36" tabindex="-1"></a>data<span class="sc">$</span>max <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(data), <span class="cf">function</span>(x){<span class="fu">max</span>(data<span class="sc">$</span>meanfreq[[x]])})</span>
<span id="cb3-37"><a href="analysis.html#cb3-37" tabindex="-1"></a>data<span class="sc">$</span>bandwidth <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(data), <span class="cf">function</span>(x){<span class="fu">max</span>(data<span class="sc">$</span>meanfreq[[x]])<span class="sc">-</span><span class="fu">min</span>(data<span class="sc">$</span>meanfreq[[x]])}) </span>
<span id="cb3-38"><a href="analysis.html#cb3-38" tabindex="-1"></a>data<span class="sc">$</span>duration <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(data), <span class="cf">function</span>(x){<span class="fu">length</span>(data<span class="sc">$</span>meanfreq[[x]])}) </span>
<span id="cb3-39"><a href="analysis.html#cb3-39" tabindex="-1"></a></span>
<span id="cb3-40"><a href="analysis.html#cb3-40" tabindex="-1"></a><span class="co">#add syllable types to data file</span></span>
<span id="cb3-41"><a href="analysis.html#cb3-41" tabindex="-1"></a>data<span class="sc">$</span>cluster_0 <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(hybrid_cut[[<span class="dv">1</span>]])</span>
<span id="cb3-42"><a href="analysis.html#cb3-42" tabindex="-1"></a>data<span class="sc">$</span>cluster_1 <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(hybrid_cut[[<span class="dv">2</span>]])</span>
<span id="cb3-43"><a href="analysis.html#cb3-43" tabindex="-1"></a>data<span class="sc">$</span>cluster_2 <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(hybrid_cut[[<span class="dv">3</span>]])</span>
<span id="cb3-44"><a href="analysis.html#cb3-44" tabindex="-1"></a>data<span class="sc">$</span>cluster_3 <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(hybrid_cut[[<span class="dv">4</span>]])</span>
<span id="cb3-45"><a href="analysis.html#cb3-45" tabindex="-1"></a>data<span class="sc">$</span>cluster_4 <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(hybrid_cut[[<span class="dv">5</span>]])</span>
<span id="cb3-46"><a href="analysis.html#cb3-46" tabindex="-1"></a></span>
<span id="cb3-47"><a href="analysis.html#cb3-47" tabindex="-1"></a><span class="co">#add year</span></span>
<span id="cb3-48"><a href="analysis.html#cb3-48" tabindex="-1"></a>data<span class="sc">$</span>year <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb3-49"><a href="analysis.html#cb3-49" tabindex="-1"></a>data<span class="sc">$</span>year[<span class="fu">which</span>(<span class="fu">substring</span>(data<span class="sc">$</span>individual, <span class="dv">1</span>, <span class="dv">4</span>) <span class="sc">==</span> <span class="st">&quot;1975&quot;</span>)] <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb3-50"><a href="analysis.html#cb3-50" tabindex="-1"></a>data<span class="sc">$</span>year[<span class="fu">which</span>(<span class="fu">substring</span>(data<span class="sc">$</span>individual, <span class="dv">1</span>, <span class="dv">4</span>) <span class="sc">==</span> <span class="st">&quot;2012&quot;</span>)] <span class="ot">&lt;-</span> <span class="dv">2</span></span>
<span id="cb3-51"><a href="analysis.html#cb3-51" tabindex="-1"></a>data<span class="sc">$</span>year[<span class="fu">which</span>(<span class="fu">substring</span>(data<span class="sc">$</span>individual, <span class="dv">1</span>, <span class="dv">4</span>) <span class="sc">==</span> <span class="st">&quot;2019&quot;</span>)] <span class="ot">&lt;-</span> <span class="dv">3</span></span>
<span id="cb3-52"><a href="analysis.html#cb3-52" tabindex="-1"></a></span>
<span id="cb3-53"><a href="analysis.html#cb3-53" tabindex="-1"></a><span class="co">#add counts</span></span>
<span id="cb3-54"><a href="analysis.html#cb3-54" tabindex="-1"></a>syl_freqs_0 <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">table</span>(data<span class="sc">$</span>cluster_0))</span>
<span id="cb3-55"><a href="analysis.html#cb3-55" tabindex="-1"></a>syl_freqs_1 <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">table</span>(data<span class="sc">$</span>cluster_1))</span>
<span id="cb3-56"><a href="analysis.html#cb3-56" tabindex="-1"></a>syl_freqs_2 <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">table</span>(data<span class="sc">$</span>cluster_2))</span>
<span id="cb3-57"><a href="analysis.html#cb3-57" tabindex="-1"></a>syl_freqs_3 <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">table</span>(data<span class="sc">$</span>cluster_3))</span>
<span id="cb3-58"><a href="analysis.html#cb3-58" tabindex="-1"></a>syl_freqs_4 <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">table</span>(data<span class="sc">$</span>cluster_4))</span>
<span id="cb3-59"><a href="analysis.html#cb3-59" tabindex="-1"></a>data<span class="sc">$</span>count_0 <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(data), <span class="cf">function</span>(x){syl_freqs_0<span class="sc">$</span>Freq[<span class="fu">which</span>(syl_freqs_0<span class="sc">$</span>Var1 <span class="sc">==</span> data<span class="sc">$</span>cluster_0[x])]})</span>
<span id="cb3-60"><a href="analysis.html#cb3-60" tabindex="-1"></a>data<span class="sc">$</span>count_1 <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(data), <span class="cf">function</span>(x){syl_freqs_1<span class="sc">$</span>Freq[<span class="fu">which</span>(syl_freqs_1<span class="sc">$</span>Var1 <span class="sc">==</span> data<span class="sc">$</span>cluster_1[x])]})</span>
<span id="cb3-61"><a href="analysis.html#cb3-61" tabindex="-1"></a>data<span class="sc">$</span>count_2 <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(data), <span class="cf">function</span>(x){syl_freqs_2<span class="sc">$</span>Freq[<span class="fu">which</span>(syl_freqs_2<span class="sc">$</span>Var1 <span class="sc">==</span> data<span class="sc">$</span>cluster_2[x])]})</span>
<span id="cb3-62"><a href="analysis.html#cb3-62" tabindex="-1"></a>data<span class="sc">$</span>count_3 <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(data), <span class="cf">function</span>(x){syl_freqs_3<span class="sc">$</span>Freq[<span class="fu">which</span>(syl_freqs_3<span class="sc">$</span>Var1 <span class="sc">==</span> data<span class="sc">$</span>cluster_3[x])]})</span>
<span id="cb3-63"><a href="analysis.html#cb3-63" tabindex="-1"></a>data<span class="sc">$</span>count_4 <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(data), <span class="cf">function</span>(x){syl_freqs_4<span class="sc">$</span>Freq[<span class="fu">which</span>(syl_freqs_4<span class="sc">$</span>Var1 <span class="sc">==</span> data<span class="sc">$</span>cluster_4[x])]})</span>
<span id="cb3-64"><a href="analysis.html#cb3-64" tabindex="-1"></a></span>
<span id="cb3-65"><a href="analysis.html#cb3-65" tabindex="-1"></a><span class="co">#add song lengths</span></span>
<span id="cb3-66"><a href="analysis.html#cb3-66" tabindex="-1"></a>song_freq_table <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">table</span>(data<span class="sc">$</span>song))</span>
<span id="cb3-67"><a href="analysis.html#cb3-67" tabindex="-1"></a>data<span class="sc">$</span>song_length <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(data), <span class="cf">function</span>(x){song_freq_table<span class="sc">$</span>Freq[<span class="fu">which</span>(song_freq_table<span class="sc">$</span>Var1 <span class="sc">==</span> data<span class="sc">$</span>song[x])]})</span>
<span id="cb3-68"><a href="analysis.html#cb3-68" tabindex="-1"></a></span>
<span id="cb3-69"><a href="analysis.html#cb3-69" tabindex="-1"></a><span class="co">#create data frame of syllable types</span></span>
<span id="cb3-70"><a href="analysis.html#cb3-70" tabindex="-1"></a>syl_types <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="dv">0</span><span class="sc">:</span><span class="dv">4</span>, <span class="cf">function</span>(x){</span>
<span id="cb3-71"><a href="analysis.html#cb3-71" tabindex="-1"></a>  col_id <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="fu">names</span>(data) <span class="sc">==</span> <span class="fu">paste0</span>(<span class="st">&quot;cluster_&quot;</span>, x))</span>
<span id="cb3-72"><a href="analysis.html#cb3-72" tabindex="-1"></a>  temp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">type =</span> <span class="fu">sort</span>(<span class="fu">unique</span>(<span class="fu">as.data.frame</span>(data)[, col_id])), <span class="at">count =</span> <span class="cn">NA</span>, <span class="at">concavity =</span> <span class="cn">NA</span>, <span class="at">excursion =</span> <span class="cn">NA</span>, <span class="at">average =</span> <span class="cn">NA</span>, <span class="at">min =</span> <span class="cn">NA</span>, <span class="at">max =</span> <span class="cn">NA</span>, <span class="at">bandwidth =</span> <span class="cn">NA</span>, <span class="at">duration =</span> <span class="cn">NA</span>)</span>
<span id="cb3-73"><a href="analysis.html#cb3-73" tabindex="-1"></a>  </span>
<span id="cb3-74"><a href="analysis.html#cb3-74" tabindex="-1"></a>  <span class="co">#get average measures for each syllable type</span></span>
<span id="cb3-75"><a href="analysis.html#cb3-75" tabindex="-1"></a>  <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(temp)){</span>
<span id="cb3-76"><a href="analysis.html#cb3-76" tabindex="-1"></a>    inds <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="fu">as.data.frame</span>(data)[, col_id] <span class="sc">==</span> temp<span class="sc">$</span>type[i])</span>
<span id="cb3-77"><a href="analysis.html#cb3-77" tabindex="-1"></a>    temp<span class="sc">$</span>count[i] <span class="ot">&lt;-</span> <span class="fu">length</span>(inds)</span>
<span id="cb3-78"><a href="analysis.html#cb3-78" tabindex="-1"></a>    temp<span class="sc">$</span>concavity[i] <span class="ot">&lt;-</span> <span class="fu">median</span>(data<span class="sc">$</span>concavity[inds])</span>
<span id="cb3-79"><a href="analysis.html#cb3-79" tabindex="-1"></a>    temp<span class="sc">$</span>excursion[i] <span class="ot">&lt;-</span> <span class="fu">median</span>(data<span class="sc">$</span>excursion[inds])</span>
<span id="cb3-80"><a href="analysis.html#cb3-80" tabindex="-1"></a>    temp<span class="sc">$</span>average[i] <span class="ot">&lt;-</span> <span class="fu">median</span>(data<span class="sc">$</span>average[inds])</span>
<span id="cb3-81"><a href="analysis.html#cb3-81" tabindex="-1"></a>    temp<span class="sc">$</span>min[i] <span class="ot">&lt;-</span> <span class="fu">median</span>(data<span class="sc">$</span>min[inds])</span>
<span id="cb3-82"><a href="analysis.html#cb3-82" tabindex="-1"></a>    temp<span class="sc">$</span>max[i] <span class="ot">&lt;-</span> <span class="fu">median</span>(data<span class="sc">$</span>max[inds])</span>
<span id="cb3-83"><a href="analysis.html#cb3-83" tabindex="-1"></a>    temp<span class="sc">$</span>bandwidth[i] <span class="ot">&lt;-</span> <span class="fu">median</span>(data<span class="sc">$</span>bandwidth[inds])</span>
<span id="cb3-84"><a href="analysis.html#cb3-84" tabindex="-1"></a>    temp<span class="sc">$</span>duration[i] <span class="ot">&lt;-</span> <span class="fu">median</span>(data<span class="sc">$</span>duration[inds])</span>
<span id="cb3-85"><a href="analysis.html#cb3-85" tabindex="-1"></a>  }</span>
<span id="cb3-86"><a href="analysis.html#cb3-86" tabindex="-1"></a>  </span>
<span id="cb3-87"><a href="analysis.html#cb3-87" tabindex="-1"></a>  <span class="fu">return</span>(temp)</span>
<span id="cb3-88"><a href="analysis.html#cb3-88" tabindex="-1"></a>})</span>
<span id="cb3-89"><a href="analysis.html#cb3-89" tabindex="-1"></a></span>
<span id="cb3-90"><a href="analysis.html#cb3-90" tabindex="-1"></a><span class="co">#save processed version of data</span></span>
<span id="cb3-91"><a href="analysis.html#cb3-91" tabindex="-1"></a>processed_data <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">data =</span> data, <span class="at">syl_types =</span> syl_types)</span>
<span id="cb3-92"><a href="analysis.html#cb3-92" tabindex="-1"></a><span class="fu">save</span>(processed_data, <span class="at">file =</span> <span class="st">&quot;data_models/processed_data.RData&quot;</span>)</span></code></pre></div>
</details>
</div>
<div id="zipfs-rank-frequency-law" class="section level2 hasAnchor" number="2.3">
<h2><span class="header-section-number">2.3</span> Zipf’s Rank-Frequency Law<a href="analysis.html#zipfs-rank-frequency-law" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Mandelbrot’s generalization of Zipf’s rank-frequency law takes the following form <span class="citation">(<a href="#ref-Mandelbrot1953">22</a>,<a href="#ref-Mandelbrot1962">23</a>)</span>:</p>
<p><span class="math display" id="eq:zipf-rf-a">\[\begin{equation}
f(r) = \frac{c}{(r + \beta)^\alpha}
\tag{2}
\end{equation}\]</span></p>
<p>where <span class="math inline">\(f(r)\)</span> is the normalized frequency at each rank <span class="math inline">\(r\)</span>, <span class="math inline">\(c\)</span> is a normalization term, and <span class="math inline">\(\alpha\)</span> and <span class="math inline">\(\beta\)</span> are parameters that control slope and convexity (respectively). According to <span class="citation">(<a href="#ref-izsak06">91</a>)</span>, the bounds of <a href="analysis.html#eq:zipf-rf-a">(2)</a> are <span class="math inline">\(\alpha &gt; 1\)</span>, <span class="math inline">\(\beta &gt; -1\)</span>, and <span class="math inline">\(c &gt; 1\)</span>. When <span class="math inline">\(\beta = 0\)</span>, this function simplifies to the original form of Zipf’s rank-frequency law: <span class="math inline">\(f(r) \propto 1/r^\alpha\)</span>.</p>
<!-- $c$ is usually a normalization constant [@izsak06; @mouillot_lepretre00; @macutek22] defined as: -->
<!-- \begin{equation} -->
<!-- c = \sum_{i=1}^{\infty}\frac{1}{(r + \beta)^\alpha} -->
<!-- (\#eq:zipf-rf-b) -->
<!-- \end{equation} -->
<!-- In practice, this form of Zipf's rank-frequency law is notoriously difficult to fit to data due to strong correlations between $\alpha$ and $\beta$, which in turn determine $c$ [@mouillot_lepretre00; @izsak06; @macutek22]. Here, I use a simplified version of \@ref(eq:zipf-rf-a) that treats $c$ as a third parameter that is estimated alongside $\alpha$ and $\beta$ [@ausloos14], as has been done in studies of chickadee calls [@hailman94; @freeberg_lucas12; @ficken_etal94, which should be interpreted as an approximation of the Zipf-Mandelbrot distribution. -->
<p>I fit <a href="analysis.html#eq:zipf-rf-a">(2)</a> to the rank-frequency distributions of the syllable classifications from each level of deep split in two batches. First, I fit all three parameters to approximate Mandelbrot’s versions of the rank-frequency law. Then, I set <span class="math inline">\(\beta = 0\)</span> to approximate Zipf’s original formulation.</p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="analysis.html#cb4-1" tabindex="-1"></a><span class="co">#load library</span></span>
<span id="cb4-2"><a href="analysis.html#cb4-2" tabindex="-1"></a><span class="fu">library</span>(brms)</span>
<span id="cb4-3"><a href="analysis.html#cb4-3" tabindex="-1"></a></span>
<span id="cb4-4"><a href="analysis.html#cb4-4" tabindex="-1"></a><span class="co">#load processed data and rename</span></span>
<span id="cb4-5"><a href="analysis.html#cb4-5" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;data_models/processed_data.RData&quot;</span>)</span>
<span id="cb4-6"><a href="analysis.html#cb4-6" tabindex="-1"></a>data <span class="ot">&lt;-</span> processed_data<span class="sc">$</span>data</span>
<span id="cb4-7"><a href="analysis.html#cb4-7" tabindex="-1"></a>syl_types <span class="ot">&lt;-</span> processed_data<span class="sc">$</span>syl_types</span>
<span id="cb4-8"><a href="analysis.html#cb4-8" tabindex="-1"></a><span class="fu">rm</span>(processed_data)</span>
<span id="cb4-9"><a href="analysis.html#cb4-9" tabindex="-1"></a></span>
<span id="cb4-10"><a href="analysis.html#cb4-10" tabindex="-1"></a><span class="co">#get frequencies for three levels of deep split</span></span>
<span id="cb4-11"><a href="analysis.html#cb4-11" tabindex="-1"></a>freqs_a <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">sort</span>(<span class="fu">table</span>(data<span class="sc">$</span>cluster_2), <span class="at">decreasing =</span> <span class="cn">TRUE</span>))<span class="sc">/</span><span class="fu">nrow</span>(data)</span>
<span id="cb4-12"><a href="analysis.html#cb4-12" tabindex="-1"></a>freqs_b <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">sort</span>(<span class="fu">table</span>(data<span class="sc">$</span>cluster_3), <span class="at">decreasing =</span> <span class="cn">TRUE</span>))<span class="sc">/</span><span class="fu">nrow</span>(data)</span>
<span id="cb4-13"><a href="analysis.html#cb4-13" tabindex="-1"></a>freqs_c <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">sort</span>(<span class="fu">table</span>(data<span class="sc">$</span>cluster_4), <span class="at">decreasing =</span> <span class="cn">TRUE</span>))<span class="sc">/</span><span class="fu">nrow</span>(data)</span>
<span id="cb4-14"><a href="analysis.html#cb4-14" tabindex="-1"></a>data_a <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">freq =</span> freqs_a, <span class="at">rank =</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(freqs_a))</span>
<span id="cb4-15"><a href="analysis.html#cb4-15" tabindex="-1"></a>data_b <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">freq =</span> freqs_b, <span class="at">rank =</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(freqs_b))</span>
<span id="cb4-16"><a href="analysis.html#cb4-16" tabindex="-1"></a>data_c <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">freq =</span> freqs_c, <span class="at">rank =</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(freqs_c))</span>
<span id="cb4-17"><a href="analysis.html#cb4-17" tabindex="-1"></a></span>
<span id="cb4-18"><a href="analysis.html#cb4-18" tabindex="-1"></a><span class="co">#set priors</span></span>
<span id="cb4-19"><a href="analysis.html#cb4-19" tabindex="-1"></a>zipf_priors <span class="ot">&lt;-</span> <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="dv">10</span>), <span class="at">lb =</span> <span class="dv">1</span>, <span class="at">nlpar =</span> <span class="st">&quot;a&quot;</span>) <span class="sc">+</span> </span>
<span id="cb4-20"><a href="analysis.html#cb4-20" tabindex="-1"></a>  <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="dv">10</span>), <span class="at">lb =</span> <span class="dv">0</span>, <span class="at">nlpar =</span> <span class="st">&quot;c&quot;</span>)</span>
<span id="cb4-21"><a href="analysis.html#cb4-21" tabindex="-1"></a>mand_priors <span class="ot">&lt;-</span> <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="dv">10</span>), <span class="at">lb =</span> <span class="dv">1</span>, <span class="at">nlpar =</span> <span class="st">&quot;a&quot;</span>) <span class="sc">+</span> </span>
<span id="cb4-22"><a href="analysis.html#cb4-22" tabindex="-1"></a>  <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="dv">10</span>), <span class="at">lb =</span> <span class="sc">-</span><span class="dv">1</span>, <span class="at">nlpar =</span> <span class="st">&quot;b&quot;</span>) <span class="sc">+</span> </span>
<span id="cb4-23"><a href="analysis.html#cb4-23" tabindex="-1"></a>  <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="dv">10</span>), <span class="at">lb =</span> <span class="dv">0</span>, <span class="at">nlpar =</span> <span class="st">&quot;c&quot;</span>)</span>
<span id="cb4-24"><a href="analysis.html#cb4-24" tabindex="-1"></a></span>
<span id="cb4-25"><a href="analysis.html#cb4-25" tabindex="-1"></a><span class="co">#fit the zipf models</span></span>
<span id="cb4-26"><a href="analysis.html#cb4-26" tabindex="-1"></a>zipf_fit_a <span class="ot">&lt;-</span> <span class="fu">brm</span>(<span class="fu">bf</span>(freq <span class="sc">~</span> c<span class="sc">/</span>(rank<span class="sc">^</span>a), a <span class="sc">+</span> c <span class="sc">~</span> <span class="dv">1</span>, <span class="at">nl =</span> <span class="cn">TRUE</span>),</span>
<span id="cb4-27"><a href="analysis.html#cb4-27" tabindex="-1"></a>                  <span class="at">data =</span> data_a, <span class="at">prior =</span> zipf_priors,</span>
<span id="cb4-28"><a href="analysis.html#cb4-28" tabindex="-1"></a>                  <span class="at">iter =</span> <span class="dv">5000</span>, <span class="at">cores =</span> <span class="dv">4</span>)</span>
<span id="cb4-29"><a href="analysis.html#cb4-29" tabindex="-1"></a>zipf_fit_b <span class="ot">&lt;-</span> <span class="fu">brm</span>(<span class="fu">bf</span>(freq <span class="sc">~</span> c<span class="sc">/</span>(rank<span class="sc">^</span>a), a <span class="sc">+</span> c <span class="sc">~</span> <span class="dv">1</span>, <span class="at">nl =</span> <span class="cn">TRUE</span>),</span>
<span id="cb4-30"><a href="analysis.html#cb4-30" tabindex="-1"></a>                  <span class="at">data =</span> data_b, <span class="at">prior =</span> zipf_priors,</span>
<span id="cb4-31"><a href="analysis.html#cb4-31" tabindex="-1"></a>                  <span class="at">iter =</span> <span class="dv">5000</span>, <span class="at">cores =</span> <span class="dv">4</span>)</span>
<span id="cb4-32"><a href="analysis.html#cb4-32" tabindex="-1"></a>zipf_fit_c <span class="ot">&lt;-</span> <span class="fu">brm</span>(<span class="fu">bf</span>(freq <span class="sc">~</span> c<span class="sc">/</span>(rank<span class="sc">^</span>a), a <span class="sc">+</span> c <span class="sc">~</span> <span class="dv">1</span>, <span class="at">nl =</span> <span class="cn">TRUE</span>),</span>
<span id="cb4-33"><a href="analysis.html#cb4-33" tabindex="-1"></a>                  <span class="at">data =</span> data_c, <span class="at">prior =</span> zipf_priors,</span>
<span id="cb4-34"><a href="analysis.html#cb4-34" tabindex="-1"></a>                  <span class="at">iter =</span> <span class="dv">5000</span>, <span class="at">cores =</span> <span class="dv">4</span>)</span>
<span id="cb4-35"><a href="analysis.html#cb4-35" tabindex="-1"></a></span>
<span id="cb4-36"><a href="analysis.html#cb4-36" tabindex="-1"></a><span class="co">#fit the mandelbrot models</span></span>
<span id="cb4-37"><a href="analysis.html#cb4-37" tabindex="-1"></a>mand_fit_a <span class="ot">&lt;-</span> <span class="fu">brm</span>(<span class="fu">bf</span>(freq <span class="sc">~</span> c<span class="sc">/</span>((rank <span class="sc">+</span> b)<span class="sc">^</span>a), a <span class="sc">+</span> b <span class="sc">+</span> c <span class="sc">~</span> <span class="dv">1</span>, <span class="at">nl =</span> <span class="cn">TRUE</span>),</span>
<span id="cb4-38"><a href="analysis.html#cb4-38" tabindex="-1"></a>                  <span class="at">data =</span> data_a, <span class="at">prior =</span> mand_priors,</span>
<span id="cb4-39"><a href="analysis.html#cb4-39" tabindex="-1"></a>                  <span class="at">iter =</span> <span class="dv">5000</span>, <span class="at">cores =</span> <span class="dv">4</span>)</span>
<span id="cb4-40"><a href="analysis.html#cb4-40" tabindex="-1"></a>mand_fit_b <span class="ot">&lt;-</span> <span class="fu">brm</span>(<span class="fu">bf</span>(freq <span class="sc">~</span> c<span class="sc">/</span>((rank <span class="sc">+</span> b)<span class="sc">^</span>a), a <span class="sc">+</span> b <span class="sc">+</span> c <span class="sc">~</span> <span class="dv">1</span>, <span class="at">nl =</span> <span class="cn">TRUE</span>),</span>
<span id="cb4-41"><a href="analysis.html#cb4-41" tabindex="-1"></a>                  <span class="at">data =</span> data_b, <span class="at">prior =</span> mand_priors,</span>
<span id="cb4-42"><a href="analysis.html#cb4-42" tabindex="-1"></a>                  <span class="at">iter =</span> <span class="dv">5000</span>, <span class="at">cores =</span> <span class="dv">4</span>)</span>
<span id="cb4-43"><a href="analysis.html#cb4-43" tabindex="-1"></a>mand_fit_c <span class="ot">&lt;-</span> <span class="fu">brm</span>(<span class="fu">bf</span>(freq <span class="sc">~</span> c<span class="sc">/</span>((rank <span class="sc">+</span> b)<span class="sc">^</span>a), a <span class="sc">+</span> b <span class="sc">+</span> c <span class="sc">~</span> <span class="dv">1</span>, <span class="at">nl =</span> <span class="cn">TRUE</span>),</span>
<span id="cb4-44"><a href="analysis.html#cb4-44" tabindex="-1"></a>                  <span class="at">data =</span> data_c, <span class="at">prior =</span> mand_priors,</span>
<span id="cb4-45"><a href="analysis.html#cb4-45" tabindex="-1"></a>                  <span class="at">iter =</span> <span class="dv">5000</span>, <span class="at">cores =</span> <span class="dv">4</span>)</span>
<span id="cb4-46"><a href="analysis.html#cb4-46" tabindex="-1"></a></span>
<span id="cb4-47"><a href="analysis.html#cb4-47" tabindex="-1"></a><span class="co">#store and save all relevant output</span></span>
<span id="cb4-48"><a href="analysis.html#cb4-48" tabindex="-1"></a>zipf_rf_models <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">data =</span> <span class="fu">list</span>(<span class="at">ds_2 =</span> data_a,</span>
<span id="cb4-49"><a href="analysis.html#cb4-49" tabindex="-1"></a>                                   <span class="at">ds_3 =</span> data_b,</span>
<span id="cb4-50"><a href="analysis.html#cb4-50" tabindex="-1"></a>                                   <span class="at">ds_4 =</span> data_c),</span>
<span id="cb4-51"><a href="analysis.html#cb4-51" tabindex="-1"></a>                       <span class="at">zipf =</span> <span class="fu">list</span>(<span class="at">ds_2 =</span> <span class="fu">summary</span>(zipf_fit_a)<span class="sc">$</span>fixed,</span>
<span id="cb4-52"><a href="analysis.html#cb4-52" tabindex="-1"></a>                                   <span class="at">ds_3 =</span> <span class="fu">summary</span>(zipf_fit_b)<span class="sc">$</span>fixed, </span>
<span id="cb4-53"><a href="analysis.html#cb4-53" tabindex="-1"></a>                                   <span class="at">ds_4 =</span> <span class="fu">summary</span>(zipf_fit_c)<span class="sc">$</span>fixed), </span>
<span id="cb4-54"><a href="analysis.html#cb4-54" tabindex="-1"></a>                       <span class="at">mand =</span> <span class="fu">list</span>(<span class="at">ds_2 =</span> <span class="fu">summary</span>(mand_fit_a)<span class="sc">$</span>fixed, </span>
<span id="cb4-55"><a href="analysis.html#cb4-55" tabindex="-1"></a>                                   <span class="at">ds_3 =</span> <span class="fu">summary</span>(mand_fit_b)<span class="sc">$</span>fixed, </span>
<span id="cb4-56"><a href="analysis.html#cb4-56" tabindex="-1"></a>                                   <span class="at">ds_4 =</span> <span class="fu">summary</span>(mand_fit_c)<span class="sc">$</span>fixed), </span>
<span id="cb4-57"><a href="analysis.html#cb4-57" tabindex="-1"></a>                       <span class="at">prior =</span> <span class="fu">prior_summary</span>(mand_fit_a),</span>
<span id="cb4-58"><a href="analysis.html#cb4-58" tabindex="-1"></a>                       <span class="at">waic =</span> <span class="fu">data.frame</span>(<span class="at">zipf =</span> <span class="fu">c</span>(<span class="fu">waic</span>(zipf_fit_a)<span class="sc">$</span>waic, <span class="fu">waic</span>(zipf_fit_b)<span class="sc">$</span>waic, <span class="fu">waic</span>(zipf_fit_c)<span class="sc">$</span>waic),</span>
<span id="cb4-59"><a href="analysis.html#cb4-59" tabindex="-1"></a>                                         <span class="at">mand =</span> <span class="fu">c</span>(<span class="fu">waic</span>(mand_fit_a)<span class="sc">$</span>waic, <span class="fu">waic</span>(mand_fit_b)<span class="sc">$</span>waic, <span class="fu">waic</span>(mand_fit_c)<span class="sc">$</span>waic)),</span>
<span id="cb4-60"><a href="analysis.html#cb4-60" tabindex="-1"></a>                       <span class="at">r2 =</span> <span class="fu">c</span>(<span class="fu">bayes_R2</span>(zipf_fit_a)[<span class="dv">1</span>], <span class="fu">bayes_R2</span>(zipf_fit_b)[<span class="dv">1</span>], <span class="fu">bayes_R2</span>(zipf_fit_c)[<span class="dv">1</span>], </span>
<span id="cb4-61"><a href="analysis.html#cb4-61" tabindex="-1"></a>                              <span class="fu">bayes_R2</span>(mand_fit_a)[<span class="dv">1</span>], <span class="fu">bayes_R2</span>(mand_fit_b)[<span class="dv">1</span>], <span class="fu">bayes_R2</span>(mand_fit_c)[<span class="dv">1</span>]))</span>
<span id="cb4-62"><a href="analysis.html#cb4-62" tabindex="-1"></a><span class="fu">save</span>(zipf_rf_models, <span class="at">file =</span> <span class="st">&quot;data_models/zipf_rf_models.RData&quot;</span>)</span></code></pre></div>
</details>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:figure-zipf-rank-small"></span>
<img src="plots/rank_count_small.png" alt="The relationship between rank (x-axis) and count (y-axis) at the intermediate granularity level (DS = 3, 596 syllable types). The blue and orange lines denote the expected distributions according to Zipf's rank-frequency law (blue) and Mandelbrot's extension of it (orange)." width="50%" />
<p class="caption">
Figure 2: The relationship between rank (x-axis) and count (y-axis) at the intermediate granularity level (DS = 3, 596 syllable types). The blue and orange lines denote the expected distributions according to Zipf’s rank-frequency law (blue) and Mandelbrot’s extension of it (orange).
</p>
</div>
<p>The observed frequency distribution is consistent with the Zipf-Mandelbrot distribution at all three levels of granularity (<span class="math inline">\(R^2 = \{0.995, 0.995, 0.995\}\)</span> at <span class="math inline">\(DS = \{2, 3, 4\}\)</span>), but is only weakly consistent with the the original form of Zipf’s rank frequency law (<span class="math inline">\(R^2 = \{0.945, 0.777, 0.523\}\)</span> at <span class="math inline">\(DS = \{2, 3, 4\}\)</span>). Figure <a href="analysis.html#fig:figure-zipf-rank-small">2</a> shows the rank-frequency distribution at the intermediate granularity level (<span class="math inline">\(DS = 3\)</span>; 596 syllable types), while all three can be seen in the <a href="https://masonyoungblood.github.io/linguistic_efficiency/supplement/supplement.html">Supplementary Information</a>. <!--$R^2$ is the most commonly reported goodness-of-fit measure for both forms of Zipf's rank-frequency law, and is generally $> 0.8$ in studies of human language [@linders_louwerse23].--> The Zipf-Mandelbrot distribution also outcompetes Zipf’s original form and has <span class="math inline">\(R^2 &gt; 0.98\)</span> when models are fit separately to the data from each year (1975, 2012, and 2019; see <a href="https://masonyoungblood.github.io/linguistic_efficiency/supplement/supplement.html">Supplementary Information</a>).</p>
<!-- The $WAIC$ values and fitted parameters for all six models can be found in the [Supplementary Information]. -->
</div>
<div id="zipfs-law-of-abbreviation" class="section level2 hasAnchor" number="2.4">
<h2><span class="header-section-number">2.4</span> Zipf’s Law of Abbreviation<a href="analysis.html#zipfs-law-of-abbreviation" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Zipf’s law of abbreviation predicts that common items will be shorter in duration than rarer ones. Rather than focusing duration alone, I explored whether the frequency of syllables is negatively correlated with four different measures of production cost: duration (ms), bandwidth (Hz), concavity (changes in the sign of the slope of the mean frequency trace/ms) and excursion (cumulative absolute change in Hz/ms). <!--As mentioned in [Clustering], concavity and excursion are both indicators of syllable complexity [@Ju2019; @podosFinescaleBroadlyApplicable2016].--></p>
<p>For each level of deep split and each measure of production cost, I constructed a log-normal model with the measure in question as the outcome variable, count as the predictor variable, and syllable type as a varying intercept. The alternative formulation, a Poisson model with count as the outcome variable, does not allow for the correct random effects structure (e.g. counts are identical across observations of the same syllable type). <!--Zipf's law of abbreviation predicts that there will be a strong effect of count on each measure of production cost.--></p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="analysis.html#cb5-1" tabindex="-1"></a><span class="co">#load packages</span></span>
<span id="cb5-2"><a href="analysis.html#cb5-2" tabindex="-1"></a><span class="fu">library</span>(brms)</span>
<span id="cb5-3"><a href="analysis.html#cb5-3" tabindex="-1"></a></span>
<span id="cb5-4"><a href="analysis.html#cb5-4" tabindex="-1"></a><span class="co">#load processed data and rename</span></span>
<span id="cb5-5"><a href="analysis.html#cb5-5" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;data_models/processed_data.RData&quot;</span>)</span>
<span id="cb5-6"><a href="analysis.html#cb5-6" tabindex="-1"></a>data <span class="ot">&lt;-</span> processed_data<span class="sc">$</span>data</span>
<span id="cb5-7"><a href="analysis.html#cb5-7" tabindex="-1"></a></span>
<span id="cb5-8"><a href="analysis.html#cb5-8" tabindex="-1"></a><span class="co">#set priors, a has lower bound on intercept (because outcome is only positive) while b does not</span></span>
<span id="cb5-9"><a href="analysis.html#cb5-9" tabindex="-1"></a>zipf_priors_a <span class="ot">&lt;-</span> <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="dv">4</span>), <span class="at">lb =</span> <span class="dv">0</span>, <span class="at">class =</span> <span class="st">&quot;Intercept&quot;</span>) <span class="sc">+</span> </span>
<span id="cb5-10"><a href="analysis.html#cb5-10" tabindex="-1"></a>  <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> <span class="st">&quot;b&quot;</span>) <span class="sc">+</span> </span>
<span id="cb5-11"><a href="analysis.html#cb5-11" tabindex="-1"></a>  <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">lb =</span> <span class="dv">0</span>, <span class="at">class =</span> <span class="st">&quot;sd&quot;</span>) <span class="sc">+</span> </span>
<span id="cb5-12"><a href="analysis.html#cb5-12" tabindex="-1"></a>  <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">lb =</span> <span class="dv">0</span>, <span class="at">class =</span> <span class="st">&quot;sigma&quot;</span>)</span>
<span id="cb5-13"><a href="analysis.html#cb5-13" tabindex="-1"></a>zipf_priors_b <span class="ot">&lt;-</span> <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="dv">4</span>), <span class="at">class =</span> <span class="st">&quot;Intercept&quot;</span>) <span class="sc">+</span> </span>
<span id="cb5-14"><a href="analysis.html#cb5-14" tabindex="-1"></a>  <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">class =</span> <span class="st">&quot;b&quot;</span>) <span class="sc">+</span> </span>
<span id="cb5-15"><a href="analysis.html#cb5-15" tabindex="-1"></a>  <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">lb =</span> <span class="dv">0</span>, <span class="at">class =</span> <span class="st">&quot;sd&quot;</span>) <span class="sc">+</span> </span>
<span id="cb5-16"><a href="analysis.html#cb5-16" tabindex="-1"></a>  <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">lb =</span> <span class="dv">0</span>, <span class="at">class =</span> <span class="st">&quot;sigma&quot;</span>)</span>
<span id="cb5-17"><a href="analysis.html#cb5-17" tabindex="-1"></a></span>
<span id="cb5-18"><a href="analysis.html#cb5-18" tabindex="-1"></a><span class="co">#run duration model across all three deep split values</span></span>
<span id="cb5-19"><a href="analysis.html#cb5-19" tabindex="-1"></a>duration_model_2 <span class="ot">&lt;-</span> <span class="fu">brm</span>(<span class="fu">log</span>(duration) <span class="sc">~</span> <span class="fu">scale</span>(count_2) <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>cluster_2), <span class="at">data =</span> data, <span class="at">prior =</span> zipf_priors_a, <span class="at">iter =</span> <span class="dv">5000</span>, <span class="at">cores =</span> <span class="dv">4</span>)</span>
<span id="cb5-20"><a href="analysis.html#cb5-20" tabindex="-1"></a>duration_model_3 <span class="ot">&lt;-</span> <span class="fu">brm</span>(<span class="fu">log</span>(duration) <span class="sc">~</span> <span class="fu">scale</span>(count_3) <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>cluster_3), <span class="at">data =</span> data, <span class="at">prior =</span> zipf_priors_a, <span class="at">iter =</span> <span class="dv">5000</span>, <span class="at">cores =</span> <span class="dv">4</span>)</span>
<span id="cb5-21"><a href="analysis.html#cb5-21" tabindex="-1"></a>duration_model_4 <span class="ot">&lt;-</span> <span class="fu">brm</span>(<span class="fu">log</span>(duration) <span class="sc">~</span> <span class="fu">scale</span>(count_4) <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>cluster_4), <span class="at">data =</span> data, <span class="at">prior =</span> zipf_priors_a, <span class="at">iter =</span> <span class="dv">5000</span>, <span class="at">cores =</span> <span class="dv">4</span>)</span>
<span id="cb5-22"><a href="analysis.html#cb5-22" tabindex="-1"></a></span>
<span id="cb5-23"><a href="analysis.html#cb5-23" tabindex="-1"></a><span class="co">#run bandwidth model across all three deep split values</span></span>
<span id="cb5-24"><a href="analysis.html#cb5-24" tabindex="-1"></a>bandwidth_model_2 <span class="ot">&lt;-</span> <span class="fu">brm</span>(<span class="fu">log</span>(bandwidth) <span class="sc">~</span> <span class="fu">scale</span>(count_2) <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>cluster_2), <span class="at">data =</span> data, <span class="at">prior =</span> zipf_priors_a, <span class="at">iter =</span> <span class="dv">5000</span>, <span class="at">cores =</span> <span class="dv">4</span>)</span>
<span id="cb5-25"><a href="analysis.html#cb5-25" tabindex="-1"></a>bandwidth_model_3 <span class="ot">&lt;-</span> <span class="fu">brm</span>(<span class="fu">log</span>(bandwidth) <span class="sc">~</span> <span class="fu">scale</span>(count_3) <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>cluster_3), <span class="at">data =</span> data, <span class="at">prior =</span> zipf_priors_a, <span class="at">iter =</span> <span class="dv">5000</span>, <span class="at">cores =</span> <span class="dv">4</span>)</span>
<span id="cb5-26"><a href="analysis.html#cb5-26" tabindex="-1"></a>bandwidth_model_4 <span class="ot">&lt;-</span> <span class="fu">brm</span>(<span class="fu">log</span>(bandwidth) <span class="sc">~</span> <span class="fu">scale</span>(count_4) <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>cluster_4), <span class="at">data =</span> data, <span class="at">prior =</span> zipf_priors_a, <span class="at">iter =</span> <span class="dv">5000</span>, <span class="at">cores =</span> <span class="dv">4</span>)</span>
<span id="cb5-27"><a href="analysis.html#cb5-27" tabindex="-1"></a></span>
<span id="cb5-28"><a href="analysis.html#cb5-28" tabindex="-1"></a><span class="co">#run concavity model across all three deep split values, with 0.0001 added to avoid log-transformation issues</span></span>
<span id="cb5-29"><a href="analysis.html#cb5-29" tabindex="-1"></a>concavity_model_2 <span class="ot">&lt;-</span> <span class="fu">brm</span>(<span class="fu">log</span>(concavity <span class="sc">+</span> <span class="fl">0.0001</span>) <span class="sc">~</span> <span class="fu">scale</span>(count_2) <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>cluster_2), <span class="at">data =</span> data, <span class="at">prior =</span> zipf_priors_b, <span class="at">iter =</span> <span class="dv">5000</span>, <span class="at">cores =</span> <span class="dv">4</span>)</span>
<span id="cb5-30"><a href="analysis.html#cb5-30" tabindex="-1"></a>concavity_model_3 <span class="ot">&lt;-</span> <span class="fu">brm</span>(<span class="fu">log</span>(concavity <span class="sc">+</span> <span class="fl">0.0001</span>) <span class="sc">~</span> <span class="fu">scale</span>(count_3) <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>cluster_3), <span class="at">data =</span> data, <span class="at">prior =</span> zipf_priors_b, <span class="at">iter =</span> <span class="dv">5000</span>, <span class="at">cores =</span> <span class="dv">4</span>)</span>
<span id="cb5-31"><a href="analysis.html#cb5-31" tabindex="-1"></a>concavity_model_4 <span class="ot">&lt;-</span> <span class="fu">brm</span>(<span class="fu">log</span>(concavity <span class="sc">+</span> <span class="fl">0.0001</span>) <span class="sc">~</span> <span class="fu">scale</span>(count_4) <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>cluster_4), <span class="at">data =</span> data, <span class="at">prior =</span> zipf_priors_b, <span class="at">iter =</span> <span class="dv">5000</span>, <span class="at">cores =</span> <span class="dv">4</span>)</span>
<span id="cb5-32"><a href="analysis.html#cb5-32" tabindex="-1"></a></span>
<span id="cb5-33"><a href="analysis.html#cb5-33" tabindex="-1"></a><span class="co">#run excursion model across all three deep split values, with 0.0001 added to avoid log-transformation issues</span></span>
<span id="cb5-34"><a href="analysis.html#cb5-34" tabindex="-1"></a>excursion_model_2 <span class="ot">&lt;-</span> <span class="fu">brm</span>(<span class="fu">log</span>(excursion <span class="sc">+</span> <span class="fl">0.0001</span>) <span class="sc">~</span> <span class="fu">scale</span>(count_2) <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>cluster_2), <span class="at">data =</span> data, <span class="at">prior =</span> zipf_priors_b, <span class="at">iter =</span> <span class="dv">5000</span>, <span class="at">cores =</span> <span class="dv">4</span>)</span>
<span id="cb5-35"><a href="analysis.html#cb5-35" tabindex="-1"></a>excursion_model_3 <span class="ot">&lt;-</span> <span class="fu">brm</span>(<span class="fu">log</span>(excursion <span class="sc">+</span> <span class="fl">0.0001</span>) <span class="sc">~</span> <span class="fu">scale</span>(count_3) <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>cluster_3), <span class="at">data =</span> data, <span class="at">prior =</span> zipf_priors_b, <span class="at">iter =</span> <span class="dv">5000</span>, <span class="at">cores =</span> <span class="dv">4</span>)</span>
<span id="cb5-36"><a href="analysis.html#cb5-36" tabindex="-1"></a>excursion_model_4 <span class="ot">&lt;-</span> <span class="fu">brm</span>(<span class="fu">log</span>(excursion <span class="sc">+</span> <span class="fl">0.0001</span>) <span class="sc">~</span> <span class="fu">scale</span>(count_4) <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>cluster_4), <span class="at">data =</span> data, <span class="at">prior =</span> zipf_priors_b, <span class="at">iter =</span> <span class="dv">5000</span>, <span class="at">cores =</span> <span class="dv">4</span>)</span>
<span id="cb5-37"><a href="analysis.html#cb5-37" tabindex="-1"></a></span>
<span id="cb5-38"><a href="analysis.html#cb5-38" tabindex="-1"></a><span class="co">#combine into object that contains model estimates, diagnostics, and prior specifications, and then save</span></span>
<span id="cb5-39"><a href="analysis.html#cb5-39" tabindex="-1"></a>zipf_models <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">duration =</span> <span class="fu">list</span>(<span class="fu">summary</span>(duration_model_2)<span class="sc">$</span>fixed, <span class="fu">summary</span>(duration_model_3)<span class="sc">$</span>fixed, <span class="fu">summary</span>(duration_model_4)<span class="sc">$</span>fixed, <span class="fu">prior_summary</span>(duration_model_2)),</span>
<span id="cb5-40"><a href="analysis.html#cb5-40" tabindex="-1"></a>                    <span class="at">bandwidth =</span> <span class="fu">list</span>(<span class="fu">summary</span>(bandwidth_model_2)<span class="sc">$</span>fixed, <span class="fu">summary</span>(bandwidth_model_3)<span class="sc">$</span>fixed, <span class="fu">summary</span>(bandwidth_model_4)<span class="sc">$</span>fixed, <span class="fu">prior_summary</span>(bandwidth_model_2)),</span>
<span id="cb5-41"><a href="analysis.html#cb5-41" tabindex="-1"></a>                    <span class="at">concavity =</span> <span class="fu">list</span>(<span class="fu">summary</span>(concavity_model_2)<span class="sc">$</span>fixed, <span class="fu">summary</span>(concavity_model_3)<span class="sc">$</span>fixed, <span class="fu">summary</span>(concavity_model_4)<span class="sc">$</span>fixed, <span class="fu">prior_summary</span>(concavity_model_2)),</span>
<span id="cb5-42"><a href="analysis.html#cb5-42" tabindex="-1"></a>                    <span class="at">excursion =</span> <span class="fu">list</span>(<span class="fu">summary</span>(excursion_model_2)<span class="sc">$</span>fixed, <span class="fu">summary</span>(excursion_model_3)<span class="sc">$</span>fixed, <span class="fu">summary</span>(excursion_model_4)<span class="sc">$</span>fixed, <span class="fu">prior_summary</span>(excursion_model_2)))</span>
<span id="cb5-43"><a href="analysis.html#cb5-43" tabindex="-1"></a><span class="fu">save</span>(zipf_models, <span class="at">file =</span> <span class="st">&quot;data_models/zipf_models.RData&quot;</span>)</span></code></pre></div>
</details>
<div class="tabwid"><style>.cl-ceec3364{table-layout:auto;}.cl-cee84556{font-family:'Helvetica';font-size:11pt;font-weight:bold;font-style:normal;text-decoration:none;color:rgba(0, 0, 0, 1.00);background-color:transparent;}.cl-cee8456a{font-family:'Helvetica';font-size:11pt;font-weight:normal;font-style:normal;text-decoration:none;color:rgba(0, 0, 0, 1.00);background-color:transparent;}.cl-ceea1994{margin:0;text-align:left;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-ceea199e{margin:0;text-align:right;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-ceea2614{background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-ceea2615{background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-ceea2616{background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-ceea2617{background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-ceea261e{background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-ceea261f{background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-ceea2620{background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-ceea2621{background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}</style><table data-quarto-disable-processing='true' class='cl-ceec3364'>

<caption style="display:table-caption;margin:0pt;text-align:center;border-bottom: 0.00pt solid transparent;border-top: 0.00pt solid transparent;border-left: 0.00pt solid transparent;border-right: 0.00pt solid transparent;padding-top:3pt;padding-bottom:3pt;padding-left:3pt;padding-right:3pt;line-height: 1;background-color:transparent;">
<span id="tab:zipf-abbr-tables">Table 1: </span><span>The estimated effect of count on each measure of production cost, using the syllable classifications from each level of deep split. 95% credible intervals that do not overlap with 0 are marked with an asterisk.</span>
</caption>

<thead><tr style="overflow-wrap:break-word;"><th class="cl-ceea2614"><p class="cl-ceea1994"><span class="cl-cee84556">Model</span></p></th><th class="cl-ceea2615"><p class="cl-ceea199e"><span class="cl-cee84556">DS</span></p></th><th class="cl-ceea2615"><p class="cl-ceea199e"><span class="cl-cee84556">Est.</span></p></th><th class="cl-ceea2615"><p class="cl-ceea199e"><span class="cl-cee84556">Err.</span></p></th><th class="cl-ceea2615"><p class="cl-ceea199e"><span class="cl-cee84556">2.5%</span></p></th><th class="cl-ceea2615"><p class="cl-ceea199e"><span class="cl-cee84556">97.5%</span></p></th><th class="cl-ceea2614"><p class="cl-ceea1994"><span class="cl-cee84556"> </span></p></th></tr></thead><tbody><tr style="overflow-wrap:break-word;"><td  rowspan="3"class="cl-ceea2616"><p class="cl-ceea1994"><span class="cl-cee8456a">duration ~ count</span></p></td><td class="cl-ceea2617"><p class="cl-ceea199e"><span class="cl-cee8456a">2</span></p></td><td class="cl-ceea2617"><p class="cl-ceea199e"><span class="cl-cee8456a">-0.36</span></p></td><td class="cl-ceea2617"><p class="cl-ceea199e"><span class="cl-cee8456a">0.14</span></p></td><td class="cl-ceea2617"><p class="cl-ceea199e"><span class="cl-cee8456a">-0.63</span></p></td><td class="cl-ceea2617"><p class="cl-ceea199e"><span class="cl-cee8456a">-0.10</span></p></td><td class="cl-ceea2616"><p class="cl-ceea1994"><span class="cl-cee8456a">*</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-ceea261f"><p class="cl-ceea199e"><span class="cl-cee8456a">3</span></p></td><td class="cl-ceea261f"><p class="cl-ceea199e"><span class="cl-cee8456a">-0.46</span></p></td><td class="cl-ceea261f"><p class="cl-ceea199e"><span class="cl-cee8456a">0.06</span></p></td><td class="cl-ceea261f"><p class="cl-ceea199e"><span class="cl-cee8456a">-0.58</span></p></td><td class="cl-ceea261f"><p class="cl-ceea199e"><span class="cl-cee8456a">-0.34</span></p></td><td class="cl-ceea261e"><p class="cl-ceea1994"><span class="cl-cee8456a">*</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-ceea261f"><p class="cl-ceea199e"><span class="cl-cee8456a">4</span></p></td><td class="cl-ceea261f"><p class="cl-ceea199e"><span class="cl-cee8456a">-0.42</span></p></td><td class="cl-ceea261f"><p class="cl-ceea199e"><span class="cl-cee8456a">0.03</span></p></td><td class="cl-ceea261f"><p class="cl-ceea199e"><span class="cl-cee8456a">-0.48</span></p></td><td class="cl-ceea261f"><p class="cl-ceea199e"><span class="cl-cee8456a">-0.35</span></p></td><td class="cl-ceea261e"><p class="cl-ceea1994"><span class="cl-cee8456a">*</span></p></td></tr><tr style="overflow-wrap:break-word;"><td  rowspan="3"class="cl-ceea261e"><p class="cl-ceea1994"><span class="cl-cee8456a">bandwidth ~ count</span></p></td><td class="cl-ceea261f"><p class="cl-ceea199e"><span class="cl-cee8456a">2</span></p></td><td class="cl-ceea261f"><p class="cl-ceea199e"><span class="cl-cee8456a">-0.55</span></p></td><td class="cl-ceea261f"><p class="cl-ceea199e"><span class="cl-cee8456a">0.11</span></p></td><td class="cl-ceea261f"><p class="cl-ceea199e"><span class="cl-cee8456a">-0.75</span></p></td><td class="cl-ceea261f"><p class="cl-ceea199e"><span class="cl-cee8456a">-0.34</span></p></td><td class="cl-ceea261e"><p class="cl-ceea1994"><span class="cl-cee8456a">*</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-ceea261f"><p class="cl-ceea199e"><span class="cl-cee8456a">3</span></p></td><td class="cl-ceea261f"><p class="cl-ceea199e"><span class="cl-cee8456a">-0.68</span></p></td><td class="cl-ceea261f"><p class="cl-ceea199e"><span class="cl-cee8456a">0.06</span></p></td><td class="cl-ceea261f"><p class="cl-ceea199e"><span class="cl-cee8456a">-0.79</span></p></td><td class="cl-ceea261f"><p class="cl-ceea199e"><span class="cl-cee8456a">-0.57</span></p></td><td class="cl-ceea261e"><p class="cl-ceea1994"><span class="cl-cee8456a">*</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-ceea261f"><p class="cl-ceea199e"><span class="cl-cee8456a">4</span></p></td><td class="cl-ceea261f"><p class="cl-ceea199e"><span class="cl-cee8456a">-0.64</span></p></td><td class="cl-ceea261f"><p class="cl-ceea199e"><span class="cl-cee8456a">0.03</span></p></td><td class="cl-ceea261f"><p class="cl-ceea199e"><span class="cl-cee8456a">-0.69</span></p></td><td class="cl-ceea261f"><p class="cl-ceea199e"><span class="cl-cee8456a">-0.58</span></p></td><td class="cl-ceea261e"><p class="cl-ceea1994"><span class="cl-cee8456a">*</span></p></td></tr><tr style="overflow-wrap:break-word;"><td  rowspan="3"class="cl-ceea261e"><p class="cl-ceea1994"><span class="cl-cee8456a">concavity ~ count</span></p></td><td class="cl-ceea261f"><p class="cl-ceea199e"><span class="cl-cee8456a">2</span></p></td><td class="cl-ceea261f"><p class="cl-ceea199e"><span class="cl-cee8456a">-0.02</span></p></td><td class="cl-ceea261f"><p class="cl-ceea199e"><span class="cl-cee8456a">0.10</span></p></td><td class="cl-ceea261f"><p class="cl-ceea199e"><span class="cl-cee8456a">-0.21</span></p></td><td class="cl-ceea261f"><p class="cl-ceea199e"><span class="cl-cee8456a">0.17</span></p></td><td class="cl-ceea261e"><p class="cl-ceea1994"><span class="cl-cee8456a"></span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-ceea261f"><p class="cl-ceea199e"><span class="cl-cee8456a">3</span></p></td><td class="cl-ceea261f"><p class="cl-ceea199e"><span class="cl-cee8456a">-0.06</span></p></td><td class="cl-ceea261f"><p class="cl-ceea199e"><span class="cl-cee8456a">0.04</span></p></td><td class="cl-ceea261f"><p class="cl-ceea199e"><span class="cl-cee8456a">-0.15</span></p></td><td class="cl-ceea261f"><p class="cl-ceea199e"><span class="cl-cee8456a">0.02</span></p></td><td class="cl-ceea261e"><p class="cl-ceea1994"><span class="cl-cee8456a"></span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-ceea261f"><p class="cl-ceea199e"><span class="cl-cee8456a">4</span></p></td><td class="cl-ceea261f"><p class="cl-ceea199e"><span class="cl-cee8456a">-0.07</span></p></td><td class="cl-ceea261f"><p class="cl-ceea199e"><span class="cl-cee8456a">0.03</span></p></td><td class="cl-ceea261f"><p class="cl-ceea199e"><span class="cl-cee8456a">-0.12</span></p></td><td class="cl-ceea261f"><p class="cl-ceea199e"><span class="cl-cee8456a">-0.02</span></p></td><td class="cl-ceea261e"><p class="cl-ceea1994"><span class="cl-cee8456a">*</span></p></td></tr><tr style="overflow-wrap:break-word;"><td  rowspan="3"class="cl-ceea2620"><p class="cl-ceea1994"><span class="cl-cee8456a">excursion ~ count</span></p></td><td class="cl-ceea261f"><p class="cl-ceea199e"><span class="cl-cee8456a">2</span></p></td><td class="cl-ceea261f"><p class="cl-ceea199e"><span class="cl-cee8456a">-0.26</span></p></td><td class="cl-ceea261f"><p class="cl-ceea199e"><span class="cl-cee8456a">0.07</span></p></td><td class="cl-ceea261f"><p class="cl-ceea199e"><span class="cl-cee8456a">-0.41</span></p></td><td class="cl-ceea261f"><p class="cl-ceea199e"><span class="cl-cee8456a">-0.11</span></p></td><td class="cl-ceea261e"><p class="cl-ceea1994"><span class="cl-cee8456a">*</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-ceea261f"><p class="cl-ceea199e"><span class="cl-cee8456a">3</span></p></td><td class="cl-ceea261f"><p class="cl-ceea199e"><span class="cl-cee8456a">-0.33</span></p></td><td class="cl-ceea261f"><p class="cl-ceea199e"><span class="cl-cee8456a">0.04</span></p></td><td class="cl-ceea261f"><p class="cl-ceea199e"><span class="cl-cee8456a">-0.41</span></p></td><td class="cl-ceea261f"><p class="cl-ceea199e"><span class="cl-cee8456a">-0.25</span></p></td><td class="cl-ceea261e"><p class="cl-ceea1994"><span class="cl-cee8456a">*</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-ceea2621"><p class="cl-ceea199e"><span class="cl-cee8456a">4</span></p></td><td class="cl-ceea2621"><p class="cl-ceea199e"><span class="cl-cee8456a">-0.32</span></p></td><td class="cl-ceea2621"><p class="cl-ceea199e"><span class="cl-cee8456a">0.02</span></p></td><td class="cl-ceea2621"><p class="cl-ceea199e"><span class="cl-cee8456a">-0.36</span></p></td><td class="cl-ceea2621"><p class="cl-ceea199e"><span class="cl-cee8456a">-0.27</span></p></td><td class="cl-ceea2620"><p class="cl-ceea1994"><span class="cl-cee8456a">*</span></p></td></tr></tbody></table></div>
<p>Duration, bandwidth, and excursion had strong negative effects on count at all three levels of granularity (Table <a href="analysis.html#tab:zipf-abbr-tables">1</a>). Concavity had no effect on count at the first two levels of granularity, but had a very weak negative effect at the third level of granularity. These patterns are apparent in the plots of median production costs and counts in the <a href="https://masonyoungblood.github.io/linguistic_efficiency/supplement/supplement.html">Supplementary Information</a>. The results are qualitatively identical when the year of recording (1975, 2012, or 2019) is included as a varying intercept (see <a href="https://masonyoungblood.github.io/linguistic_efficiency/supplement/supplement.html">Supplementary Information</a>).</p>
</div>
<div id="menzeraths-law" class="section level2 hasAnchor" number="2.5">
<h2><span class="header-section-number">2.5</span> Menzerath’s Law<a href="analysis.html#menzeraths-law" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Menzerath’s law predicts that longer sequences will be comprised of smaller items. Importantly, Menzerath’s law is sometimes detected in random sequences from null models <span class="citation">(<a href="#ref-g_torre_etal21">92</a>–<a href="#ref-ferrer-i-cancho_etal14">94</a>)</span>. There are two sorts of null models that make sense in this context: (1) random sequences with the same number of syllables as real songs, and (2) pseudorandom sequences that match both the number of syllables and cumulative duration in time. <span class="citation">(<a href="#ref-jamesPhylogenyMechanismsShared2021">49</a>)</span> interpret the latter as approximating simple motor constraints—Menzerath’s law resulting from efficiency in production alone—where stronger effects in the real data would indicate additional mechanisms (e.g. communicative efficiency). In this study, I compare the real data against each of these to assess both the presence of Menzerath’s law and whether it is beyond what would be expected from production constraints.</p>
<p>The production constraint model of <span class="citation">(<a href="#ref-jamesPhylogenyMechanismsShared2021">49</a>)</span> works as follows. For each iteration of the model, a pseudorandom sequence was produced for each real song in the dataset. Syllables were randomly sampled (with replacement) from the population until the duration of the random sequence exceeded the duration of the real song. If the difference between the duration of the random sequence and the real song was &lt;50% of the duration of the final syllable, then the final syllable was kept in the sequence. Otherwise, it was removed. Each iteration of the model produces a set of random sequences with approximately the same distribution of durations as the real data.</p>
<p>To estimate the strength of Menzerath’s law, I constructed a log-normal model with syllable duration as the outcome variable, song length in number of syllables as the predictor variable, and the song as a varying intercept <span class="citation">(<a href="#ref-safryghinVariableExpressionLinguistic2022">95</a>)</span>. <!--The production constraint model removes variation accounted for by the year and individual bird, so song is the only varying intercept that is appropriate for comparison. -->This model was used to estimate the strength of the effect of song length on syllable duration in both the real data and 10 simulated datasets from both null models. The brm_multiple function from the brms package in R was used to fit a single model to the the 10 simulated datasets from each null model and produce a combined posterior distribution <span class="citation">(<a href="#ref-burkner17">81</a>)</span>.</p>
<p>This analysis differs from <span class="citation">(<a href="#ref-jamesPhylogenyMechanismsShared2021">49</a>)</span> in two key ways: (1) I use the actual duration of syllables rather than a single median value for each song, and (2) I compare the full posterior distributions rather than point estimates of effects. These decisions should yield more conservative conclusions about the differences between the real and simulated data. Note that syllable type is not incorporated into the modeling, so this is the only analysis that is not conducted across multiple levels of deep split.</p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="analysis.html#cb6-1" tabindex="-1"></a><span class="co">#load packages</span></span>
<span id="cb6-2"><a href="analysis.html#cb6-2" tabindex="-1"></a><span class="fu">library</span>(brms)</span>
<span id="cb6-3"><a href="analysis.html#cb6-3" tabindex="-1"></a></span>
<span id="cb6-4"><a href="analysis.html#cb6-4" tabindex="-1"></a><span class="co">#load processed data</span></span>
<span id="cb6-5"><a href="analysis.html#cb6-5" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;data_models/processed_data.RData&quot;</span>)</span>
<span id="cb6-6"><a href="analysis.html#cb6-6" tabindex="-1"></a>data <span class="ot">&lt;-</span> processed_data<span class="sc">$</span>data</span>
<span id="cb6-7"><a href="analysis.html#cb6-7" tabindex="-1"></a>syl_types <span class="ot">&lt;-</span> processed_data<span class="sc">$</span>syl_types</span>
<span id="cb6-8"><a href="analysis.html#cb6-8" tabindex="-1"></a><span class="fu">rm</span>(processed_data)</span>
<span id="cb6-9"><a href="analysis.html#cb6-9" tabindex="-1"></a></span>
<span id="cb6-10"><a href="analysis.html#cb6-10" tabindex="-1"></a><span class="co">#get unique songs and durations</span></span>
<span id="cb6-11"><a href="analysis.html#cb6-11" tabindex="-1"></a>unique_songs <span class="ot">&lt;-</span> <span class="fu">unique</span>(data<span class="sc">$</span>song)</span>
<span id="cb6-12"><a href="analysis.html#cb6-12" tabindex="-1"></a>individuals <span class="ot">&lt;-</span> <span class="fu">as.character</span>(<span class="fu">sapply</span>(unique_songs, <span class="cf">function</span>(q){data<span class="sc">$</span>individual[<span class="fu">which</span>(data<span class="sc">$</span>song <span class="sc">==</span> q)][<span class="dv">1</span>]}))</span>
<span id="cb6-13"><a href="analysis.html#cb6-13" tabindex="-1"></a>song_durs <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">sapply</span>(unique_songs, <span class="cf">function</span>(x){<span class="fu">sum</span>(data<span class="sc">$</span>duration[<span class="fu">which</span>(data<span class="sc">$</span>song <span class="sc">==</span> x)])}))</span>
<span id="cb6-14"><a href="analysis.html#cb6-14" tabindex="-1"></a></span>
<span id="cb6-15"><a href="analysis.html#cb6-15" tabindex="-1"></a><span class="co">#create 10 simulated datasets from the simple null model</span></span>
<span id="cb6-16"><a href="analysis.html#cb6-16" tabindex="-1"></a>simple_null_sims <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>, <span class="cf">function</span>(x){</span>
<span id="cb6-17"><a href="analysis.html#cb6-17" tabindex="-1"></a>  temp <span class="ot">&lt;-</span> data</span>
<span id="cb6-18"><a href="analysis.html#cb6-18" tabindex="-1"></a>  temp<span class="sc">$</span>duration <span class="ot">&lt;-</span> <span class="fu">sample</span>(temp<span class="sc">$</span>duration)</span>
<span id="cb6-19"><a href="analysis.html#cb6-19" tabindex="-1"></a>  <span class="fu">return</span>(temp)</span>
<span id="cb6-20"><a href="analysis.html#cb6-20" tabindex="-1"></a>})</span>
<span id="cb6-21"><a href="analysis.html#cb6-21" tabindex="-1"></a></span>
<span id="cb6-22"><a href="analysis.html#cb6-22" tabindex="-1"></a><span class="co">#create production constraint model based on james et al. (2021)</span></span>
<span id="cb6-23"><a href="analysis.html#cb6-23" tabindex="-1"></a>prod_null <span class="ot">&lt;-</span> <span class="cf">function</span>(){</span>
<span id="cb6-24"><a href="analysis.html#cb6-24" tabindex="-1"></a>  <span class="co">#get shuffled versions of data</span></span>
<span id="cb6-25"><a href="analysis.html#cb6-25" tabindex="-1"></a>  sampled_inds <span class="ot">&lt;-</span> <span class="fu">sample</span>(<span class="fu">nrow</span>(data))</span>
<span id="cb6-26"><a href="analysis.html#cb6-26" tabindex="-1"></a>  sampled_durs <span class="ot">&lt;-</span> data<span class="sc">$</span>duration[sampled_inds]</span>
<span id="cb6-27"><a href="analysis.html#cb6-27" tabindex="-1"></a></span>
<span id="cb6-28"><a href="analysis.html#cb6-28" tabindex="-1"></a>  <span class="co">#double everything so the algorithm can loop back around if it needs to</span></span>
<span id="cb6-29"><a href="analysis.html#cb6-29" tabindex="-1"></a>  sampled_inds <span class="ot">&lt;-</span> <span class="fu">c</span>(sampled_inds, sampled_inds)</span>
<span id="cb6-30"><a href="analysis.html#cb6-30" tabindex="-1"></a>  sampled_durs <span class="ot">&lt;-</span> <span class="fu">c</span>(sampled_durs, sampled_durs)</span>
<span id="cb6-31"><a href="analysis.html#cb6-31" tabindex="-1"></a></span>
<span id="cb6-32"><a href="analysis.html#cb6-32" tabindex="-1"></a>  <span class="co">#set starting points and sampling window for cumulative duration calculation</span></span>
<span id="cb6-33"><a href="analysis.html#cb6-33" tabindex="-1"></a>  ind <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb6-34"><a href="analysis.html#cb6-34" tabindex="-1"></a>  m <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb6-35"><a href="analysis.html#cb6-35" tabindex="-1"></a>  window <span class="ot">&lt;-</span> <span class="dv">200</span></span>
<span id="cb6-36"><a href="analysis.html#cb6-36" tabindex="-1"></a>  </span>
<span id="cb6-37"><a href="analysis.html#cb6-37" tabindex="-1"></a>  <span class="co">#create empty sequences list to fill</span></span>
<span id="cb6-38"><a href="analysis.html#cb6-38" tabindex="-1"></a>  sequences <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb6-39"><a href="analysis.html#cb6-39" tabindex="-1"></a>  </span>
<span id="cb6-40"><a href="analysis.html#cb6-40" tabindex="-1"></a>  <span class="co">#iterate through unique songs and create sequences</span></span>
<span id="cb6-41"><a href="analysis.html#cb6-41" tabindex="-1"></a>  <span class="cf">while</span>(m <span class="sc">&lt;=</span> <span class="fu">length</span>(unique_songs)){</span>
<span id="cb6-42"><a href="analysis.html#cb6-42" tabindex="-1"></a>    <span class="co">#get cumulative durations from ind to the end of the sampling window</span></span>
<span id="cb6-43"><a href="analysis.html#cb6-43" tabindex="-1"></a>    cum_durs <span class="ot">&lt;-</span> <span class="fu">cumsum</span>(sampled_durs[ind<span class="sc">:</span>(ind <span class="sc">+</span> window)])</span>
<span id="cb6-44"><a href="analysis.html#cb6-44" tabindex="-1"></a>    </span>
<span id="cb6-45"><a href="analysis.html#cb6-45" tabindex="-1"></a>    <span class="co">#set the targets as the value at which the cumulative durations surpass the desired duration</span></span>
<span id="cb6-46"><a href="analysis.html#cb6-46" tabindex="-1"></a>    rel_position <span class="ot">&lt;-</span> <span class="fu">min</span>(<span class="fu">which</span>(cum_durs <span class="sc">&gt;</span> song_durs[m])) <span class="co">#relative position in the cumulative duration window</span></span>
<span id="cb6-47"><a href="analysis.html#cb6-47" tabindex="-1"></a>    abs_position <span class="ot">&lt;-</span> ind <span class="sc">+</span> rel_position <span class="sc">-</span> <span class="dv">1</span> <span class="co">#absolute position in the sampled vectors</span></span>
<span id="cb6-48"><a href="analysis.html#cb6-48" tabindex="-1"></a>    </span>
<span id="cb6-49"><a href="analysis.html#cb6-49" tabindex="-1"></a>    <span class="co">#if syllable duration surpasses song duration at first position, do exception handling</span></span>
<span id="cb6-50"><a href="analysis.html#cb6-50" tabindex="-1"></a>    <span class="cf">if</span>(rel_position <span class="sc">==</span> <span class="dv">1</span>){</span>
<span id="cb6-51"><a href="analysis.html#cb6-51" tabindex="-1"></a>      <span class="co">#add the single-unit sequence and iterate ind</span></span>
<span id="cb6-52"><a href="analysis.html#cb6-52" tabindex="-1"></a>      sequences[[m]] <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">song =</span> unique_songs[m], <span class="at">individual =</span> individuals[m], <span class="at">duration =</span> sampled_durs[ind], <span class="at">song_length =</span> rel_position)</span>
<span id="cb6-53"><a href="analysis.html#cb6-53" tabindex="-1"></a>      ind <span class="ot">&lt;-</span> ind <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb6-54"><a href="analysis.html#cb6-54" tabindex="-1"></a>    } <span class="cf">else</span>{</span>
<span id="cb6-55"><a href="analysis.html#cb6-55" tabindex="-1"></a>      <span class="co">#if the difference between the cumulative duration at the target and the desired duration is less that half of the length of target</span></span>
<span id="cb6-56"><a href="analysis.html#cb6-56" tabindex="-1"></a>      <span class="cf">if</span>(cum_durs[rel_position] <span class="sc">-</span> song_durs[m] <span class="sc">&lt;</span> sampled_durs[abs_position]<span class="sc">/</span><span class="dv">2</span>){</span>
<span id="cb6-57"><a href="analysis.html#cb6-57" tabindex="-1"></a>        <span class="co">#add the sequence and iterate ind</span></span>
<span id="cb6-58"><a href="analysis.html#cb6-58" tabindex="-1"></a>        sequences[[m]] <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">song =</span> unique_songs[m], <span class="at">individual =</span> individuals[m], <span class="at">duration =</span> sampled_durs[ind<span class="sc">:</span>abs_position], <span class="at">song_length =</span> rel_position)</span>
<span id="cb6-59"><a href="analysis.html#cb6-59" tabindex="-1"></a>        ind <span class="ot">&lt;-</span> abs_position <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb6-60"><a href="analysis.html#cb6-60" tabindex="-1"></a>      } <span class="cf">else</span>{</span>
<span id="cb6-61"><a href="analysis.html#cb6-61" tabindex="-1"></a>        <span class="co">#otherwise, add abbreviated sequence and iterate ind</span></span>
<span id="cb6-62"><a href="analysis.html#cb6-62" tabindex="-1"></a>        sequences[[m]] <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">song =</span> unique_songs[m], <span class="at">individual =</span> individuals[m], <span class="at">duration =</span> sampled_durs[ind<span class="sc">:</span>(abs_position <span class="sc">-</span> <span class="dv">1</span>)], <span class="at">song_length =</span> rel_position <span class="sc">-</span> <span class="dv">1</span>)</span>
<span id="cb6-63"><a href="analysis.html#cb6-63" tabindex="-1"></a>        ind <span class="ot">&lt;-</span> abs_position</span>
<span id="cb6-64"><a href="analysis.html#cb6-64" tabindex="-1"></a>      }</span>
<span id="cb6-65"><a href="analysis.html#cb6-65" tabindex="-1"></a>    }</span>
<span id="cb6-66"><a href="analysis.html#cb6-66" tabindex="-1"></a>    </span>
<span id="cb6-67"><a href="analysis.html#cb6-67" tabindex="-1"></a>    <span class="co">#iterate m</span></span>
<span id="cb6-68"><a href="analysis.html#cb6-68" tabindex="-1"></a>    m <span class="ot">&lt;-</span> m <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb6-69"><a href="analysis.html#cb6-69" tabindex="-1"></a>  }</span>
<span id="cb6-70"><a href="analysis.html#cb6-70" tabindex="-1"></a>  </span>
<span id="cb6-71"><a href="analysis.html#cb6-71" tabindex="-1"></a>  <span class="co">#return combined sequences</span></span>
<span id="cb6-72"><a href="analysis.html#cb6-72" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">do.call</span>(rbind, sequences))</span>
<span id="cb6-73"><a href="analysis.html#cb6-73" tabindex="-1"></a>}</span>
<span id="cb6-74"><a href="analysis.html#cb6-74" tabindex="-1"></a></span>
<span id="cb6-75"><a href="analysis.html#cb6-75" tabindex="-1"></a><span class="co">#create 10 simulated datasets from the production constraint model</span></span>
<span id="cb6-76"><a href="analysis.html#cb6-76" tabindex="-1"></a>prod_null_sims <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>, <span class="cf">function</span>(x){<span class="fu">prod_null</span>()})</span>
<span id="cb6-77"><a href="analysis.html#cb6-77" tabindex="-1"></a></span>
<span id="cb6-78"><a href="analysis.html#cb6-78" tabindex="-1"></a><span class="co">#set priors</span></span>
<span id="cb6-79"><a href="analysis.html#cb6-79" tabindex="-1"></a>menz_priors <span class="ot">&lt;-</span> <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="dv">3</span>), <span class="at">lb =</span> <span class="dv">0</span>, <span class="at">class =</span> <span class="st">&quot;Intercept&quot;</span>) <span class="sc">+</span> </span>
<span id="cb6-80"><a href="analysis.html#cb6-80" tabindex="-1"></a>  <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.1</span>), <span class="at">class =</span> <span class="st">&quot;b&quot;</span>) <span class="sc">+</span> </span>
<span id="cb6-81"><a href="analysis.html#cb6-81" tabindex="-1"></a>  <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">lb =</span> <span class="dv">0</span>, <span class="at">class =</span> <span class="st">&quot;sd&quot;</span>) <span class="sc">+</span> </span>
<span id="cb6-82"><a href="analysis.html#cb6-82" tabindex="-1"></a>  <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">lb =</span> <span class="dv">0</span>, <span class="at">class =</span> <span class="st">&quot;sigma&quot;</span>)</span>
<span id="cb6-83"><a href="analysis.html#cb6-83" tabindex="-1"></a></span>
<span id="cb6-84"><a href="analysis.html#cb6-84" tabindex="-1"></a><span class="co">#run the actual model on the data</span></span>
<span id="cb6-85"><a href="analysis.html#cb6-85" tabindex="-1"></a>model <span class="ot">&lt;-</span> <span class="fu">brm</span>(<span class="fu">log</span>(duration) <span class="sc">~</span> <span class="fu">scale</span>(song_length) <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>song), <span class="at">data =</span> data, <span class="at">prior =</span> menz_priors, <span class="at">iter =</span> <span class="dv">5000</span>, <span class="at">cores =</span> <span class="dv">4</span>)</span>
<span id="cb6-86"><a href="analysis.html#cb6-86" tabindex="-1"></a></span>
<span id="cb6-87"><a href="analysis.html#cb6-87" tabindex="-1"></a><span class="co">#fit a single model across the 10 simple null datasets</span></span>
<span id="cb6-88"><a href="analysis.html#cb6-88" tabindex="-1"></a>simple_null_models <span class="ot">&lt;-</span> <span class="fu">brm_multiple</span>(<span class="fu">log</span>(duration) <span class="sc">~</span> <span class="fu">scale</span>(song_length) <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>song), <span class="at">data =</span> simple_null_sims, <span class="at">prior =</span> menz_priors, <span class="at">iter =</span> <span class="dv">5000</span>, <span class="at">cores =</span> <span class="dv">7</span>)</span>
<span id="cb6-89"><a href="analysis.html#cb6-89" tabindex="-1"></a></span>
<span id="cb6-90"><a href="analysis.html#cb6-90" tabindex="-1"></a><span class="co">#fit a single model across the 10 production constraint datasets</span></span>
<span id="cb6-91"><a href="analysis.html#cb6-91" tabindex="-1"></a>prod_null_models <span class="ot">&lt;-</span> <span class="fu">brm_multiple</span>(<span class="fu">log</span>(duration) <span class="sc">~</span> <span class="fu">scale</span>(song_length) <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>song), <span class="at">data =</span> prod_null_sims, <span class="at">prior =</span> menz_priors, <span class="at">iter =</span> <span class="dv">5000</span>, <span class="at">cores =</span> <span class="dv">7</span>)</span>
<span id="cb6-92"><a href="analysis.html#cb6-92" tabindex="-1"></a></span>
<span id="cb6-93"><a href="analysis.html#cb6-93" tabindex="-1"></a><span class="co">#combine into object that contains posterior samples, model estimates and diagnostics, and prior specifications, and then save</span></span>
<span id="cb6-94"><a href="analysis.html#cb6-94" tabindex="-1"></a>menz_models <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">actual =</span> <span class="fu">list</span>(<span class="at">estimates =</span> <span class="fu">summary</span>(model)<span class="sc">$</span>fixed, <span class="at">samples =</span> <span class="fu">posterior_samples</span>(model, <span class="at">pars =</span> <span class="fu">c</span>(<span class="st">&quot;scalesong_length&quot;</span>))<span class="sc">$</span>b_scalesong_length), </span>
<span id="cb6-95"><a href="analysis.html#cb6-95" tabindex="-1"></a>                    <span class="at">simple_null =</span> <span class="fu">list</span>(<span class="at">estimates =</span> <span class="fu">summary</span>(simple_null_models)<span class="sc">$</span>fixed, <span class="at">samples =</span> <span class="fu">posterior_samples</span>(simple_null_models, <span class="at">pars =</span> <span class="fu">c</span>(<span class="st">&quot;scalesong_length&quot;</span>))<span class="sc">$</span>b_scalesong_length, <span class="at">rhats =</span> simple_null_models<span class="sc">$</span>rhats), </span>
<span id="cb6-96"><a href="analysis.html#cb6-96" tabindex="-1"></a>                    <span class="at">prod_null =</span> <span class="fu">list</span>(<span class="at">estimates =</span> <span class="fu">summary</span>(prod_null_models)<span class="sc">$</span>fixed, <span class="at">samples =</span> <span class="fu">posterior_samples</span>(prod_null_models, <span class="at">pars =</span> <span class="fu">c</span>(<span class="st">&quot;scalesong_length&quot;</span>))<span class="sc">$</span>b_scalesong_length, <span class="at">rhats =</span> prod_null_models<span class="sc">$</span>rhats), </span>
<span id="cb6-97"><a href="analysis.html#cb6-97" tabindex="-1"></a>                    <span class="at">prior =</span> <span class="fu">prior_summary</span>(model))</span>
<span id="cb6-98"><a href="analysis.html#cb6-98" tabindex="-1"></a><span class="fu">save</span>(menz_models, <span class="at">file =</span> <span class="st">&quot;data_models/menz_models.RData&quot;</span>)</span></code></pre></div>
</details>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:figure-menzerath"></span>
<img src="plots/menzerath.png" alt="The left panel shows the relationship between song length in number of syllables (x-axis) and syllable duration (y-axis), with a best fit line from the fitted lognormal model. The right panel shows the posterior distribution of the effect of song length on duration for the real data (orange) compared to 10 simulated datasets from the simple (blue) and production (green) null models. The combined posteriors for the simulated datasets are based on 2,500 posterior samples from each of the 10 models." width="50%" />
<p class="caption">
Figure 3: The left panel shows the relationship between song length in number of syllables (x-axis) and syllable duration (y-axis), with a best fit line from the fitted lognormal model. The right panel shows the posterior distribution of the effect of song length on duration for the real data (orange) compared to 10 simulated datasets from the simple (blue) and production (green) null models. The combined posteriors for the simulated datasets are based on 2,500 posterior samples from each of the 10 models.
</p>
</div>
<p>The results of the log-normal model indicate that song length has a negative effect on syllable duration (Mean Estimate: -0.052; 95% CI: [-0.066, -0.038]) (left panel of Figure <a href="analysis.html#fig:figure-menzerath">3</a>). The posterior distribution for this effect from the model fit to the actual data (orange) is more negative than the posterior distributions from 10 simulated datasets from the production constraint null model of <span class="citation">(<a href="#ref-jamesPhylogenyMechanismsShared2021">49</a>)</span> (green) and a simple random null model (blue), although there is notable overlap between the lower tail of the production constraing model and the upper tail of the actual data (right panel of Figure <a href="analysis.html#fig:figure-menzerath">3</a>). The negative effect of song length on syllable duration persists when the year of recording (1975, 2012, or 2019) is included as a varying intercept (see <a href="https://masonyoungblood.github.io/linguistic_efficiency/supplement/supplement.html">Supplementary Information</a>).</p>
<!-- There are two songs in Figure \@ref(fig:figure-menzerath) that are much longer than the others (118 and 105 syllables). These are likely concatenated songs, which occur when males sing multiple songs in quick succession during courtship displays [@Ciaburri2019]. -->
</div>
<div id="small-worldness-index" class="section level2 hasAnchor" number="2.6">
<h2><span class="header-section-number">2.6</span> Small-Worldness Index<a href="analysis.html#small-worldness-index" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The small-worldness index (<span class="math inline">\(SWI\)</span>) is a ratio based on the clustering coefficient (<span class="math inline">\(C\)</span>) and average path length (<span class="math inline">\(L\)</span>) <span class="citation">(<a href="#ref-humphriesNetworkSmallworldnessQuantitative2008">52</a>)</span>:</p>
<p><span class="math display" id="eq:swi">\[\begin{equation}
SWI = \frac{C/C_{rand}}{L/L_{rand}}
\tag{3}
\end{equation}\]</span></p>
<p>where <span class="math inline">\(C_{rand}\)</span> and <span class="math inline">\(L_{rand}\)</span> are calculated from random networks of the same size as the real network. Values of <span class="math inline">\(SWI &gt; 1\)</span> are consistent with small-world structure <span class="citation">(<a href="#ref-humphriesNetworkSmallworldnessQuantitative2008">52</a>)</span>, which is thought to reflect general systematic structure and thus compressibility <span class="citation">(<a href="#ref-raviv_etal21">13</a>,<a href="#ref-allenNetworkAnalysisReveals2019">24</a>)</span>.</p>
<p>In this study, I followed <span class="citation">(<a href="#ref-allenNetworkAnalysisReveals2019">24</a>)</span> in calculating <span class="math inline">\(SWI\)</span> from the unweighted directed network of all syllable transitions in the population (lower panel of Figure <a href="analysis.html#fig:figure-small-world">4</a>A) to allow comparison with previous studies of non-human song. <!--Note that I am using the abbreviation $SWI$ rather than the more commonly used $S$ to differentiate it from entropy ($\hat{S}$, see [Mutual Information]).-->Importantly, if the frequency distribution of syllable types is very skewed, then random sequences could exhibit small-world structure simply because of clustering around common types. To avoid this confound, <span class="math inline">\(SWI\)</span> was calculated 1,000 times from the real data and 1,000 times from random sequences with the same distribution of types. <!--Variation in observed $SWI$ is due to $C_{rand}$ and $L_{rand}$ being computed from different random networks. It is also possible to compute S with a weighted undirected network [@muldoonSmallworldPropensityWeighted2016], but I opted for the unweighted form to allow for comparison with other whale and birdsong studies.--></p>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="analysis.html#cb7-1" tabindex="-1"></a><span class="co">#load library</span></span>
<span id="cb7-2"><a href="analysis.html#cb7-2" tabindex="-1"></a><span class="fu">library</span>(igraph)</span>
<span id="cb7-3"><a href="analysis.html#cb7-3" tabindex="-1"></a></span>
<span id="cb7-4"><a href="analysis.html#cb7-4" tabindex="-1"></a><span class="co">#load processed data</span></span>
<span id="cb7-5"><a href="analysis.html#cb7-5" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;data_models/processed_data.RData&quot;</span>)</span>
<span id="cb7-6"><a href="analysis.html#cb7-6" tabindex="-1"></a>data <span class="ot">&lt;-</span> processed_data<span class="sc">$</span>data</span>
<span id="cb7-7"><a href="analysis.html#cb7-7" tabindex="-1"></a>syl_types <span class="ot">&lt;-</span> processed_data<span class="sc">$</span>syl_types</span>
<span id="cb7-8"><a href="analysis.html#cb7-8" tabindex="-1"></a><span class="fu">rm</span>(processed_data)</span>
<span id="cb7-9"><a href="analysis.html#cb7-9" tabindex="-1"></a></span>
<span id="cb7-10"><a href="analysis.html#cb7-10" tabindex="-1"></a><span class="co">#get number of unique songs</span></span>
<span id="cb7-11"><a href="analysis.html#cb7-11" tabindex="-1"></a>unique_songs <span class="ot">&lt;-</span> <span class="fu">unique</span>(data<span class="sc">$</span>song)</span>
<span id="cb7-12"><a href="analysis.html#cb7-12" tabindex="-1"></a></span>
<span id="cb7-13"><a href="analysis.html#cb7-13" tabindex="-1"></a><span class="co">#set number of iterations</span></span>
<span id="cb7-14"><a href="analysis.html#cb7-14" tabindex="-1"></a>iters <span class="ot">&lt;-</span> <span class="dv">1000</span></span>
<span id="cb7-15"><a href="analysis.html#cb7-15" tabindex="-1"></a></span>
<span id="cb7-16"><a href="analysis.html#cb7-16" tabindex="-1"></a><span class="co">#compute swi at each value of deep split</span></span>
<span id="cb7-17"><a href="analysis.html#cb7-17" tabindex="-1"></a>small_world <span class="ot">&lt;-</span> parallel<span class="sc">::</span><span class="fu">mclapply</span>(<span class="dv">0</span><span class="sc">:</span><span class="dv">4</span>, <span class="cf">function</span>(ds){</span>
<span id="cb7-18"><a href="analysis.html#cb7-18" tabindex="-1"></a>  <span class="co">#get column ID corresponding to deep split value</span></span>
<span id="cb7-19"><a href="analysis.html#cb7-19" tabindex="-1"></a>  <span class="cf">if</span>(ds <span class="sc">==</span> <span class="dv">0</span>){syls <span class="ot">&lt;-</span> data<span class="sc">$</span>cluster_0}</span>
<span id="cb7-20"><a href="analysis.html#cb7-20" tabindex="-1"></a>  <span class="cf">if</span>(ds <span class="sc">==</span> <span class="dv">1</span>){syls <span class="ot">&lt;-</span> data<span class="sc">$</span>cluster_1}</span>
<span id="cb7-21"><a href="analysis.html#cb7-21" tabindex="-1"></a>  <span class="cf">if</span>(ds <span class="sc">==</span> <span class="dv">2</span>){syls <span class="ot">&lt;-</span> data<span class="sc">$</span>cluster_2}</span>
<span id="cb7-22"><a href="analysis.html#cb7-22" tabindex="-1"></a>  <span class="cf">if</span>(ds <span class="sc">==</span> <span class="dv">3</span>){syls <span class="ot">&lt;-</span> data<span class="sc">$</span>cluster_3}</span>
<span id="cb7-23"><a href="analysis.html#cb7-23" tabindex="-1"></a>  <span class="cf">if</span>(ds <span class="sc">==</span> <span class="dv">4</span>){syls <span class="ot">&lt;-</span> data<span class="sc">$</span>cluster_4}</span>
<span id="cb7-24"><a href="analysis.html#cb7-24" tabindex="-1"></a></span>
<span id="cb7-25"><a href="analysis.html#cb7-25" tabindex="-1"></a>  <span class="co">#extract all of the structured directed sequences</span></span>
<span id="cb7-26"><a href="analysis.html#cb7-26" tabindex="-1"></a>  directed_sequences <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, <span class="fu">lapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(unique_songs), <span class="cf">function</span>(i){</span>
<span id="cb7-27"><a href="analysis.html#cb7-27" tabindex="-1"></a>    sequence <span class="ot">&lt;-</span> syls[<span class="fu">which</span>(data<span class="sc">$</span>song <span class="sc">==</span> unique_songs[i])]</span>
<span id="cb7-28"><a href="analysis.html#cb7-28" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">matrix</span>(<span class="fu">c</span>(sequence[<span class="dv">1</span><span class="sc">:</span>(<span class="fu">length</span>(sequence)<span class="sc">-</span><span class="dv">1</span>)], sequence[<span class="dv">2</span><span class="sc">:</span><span class="fu">length</span>(sequence)]), <span class="at">ncol =</span> <span class="dv">2</span>))</span>
<span id="cb7-29"><a href="analysis.html#cb7-29" tabindex="-1"></a>  }))</span>
<span id="cb7-30"><a href="analysis.html#cb7-30" tabindex="-1"></a></span>
<span id="cb7-31"><a href="analysis.html#cb7-31" tabindex="-1"></a>  <span class="co">#convert to factor</span></span>
<span id="cb7-32"><a href="analysis.html#cb7-32" tabindex="-1"></a>  directed_sequences <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="fu">as.numeric</span>(<span class="fu">factor</span>(directed_sequences)), <span class="at">ncol =</span> <span class="dv">2</span>)</span>
<span id="cb7-33"><a href="analysis.html#cb7-33" tabindex="-1"></a></span>
<span id="cb7-34"><a href="analysis.html#cb7-34" tabindex="-1"></a>  <span class="co">#extract &quot;iters&quot; sets of shuffled sequences for null models</span></span>
<span id="cb7-35"><a href="analysis.html#cb7-35" tabindex="-1"></a>  shuffled_sequences <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="dv">1</span><span class="sc">:</span>iters, <span class="cf">function</span>(x){</span>
<span id="cb7-36"><a href="analysis.html#cb7-36" tabindex="-1"></a>    shuffled_syls <span class="ot">&lt;-</span> <span class="fu">sample</span>(syls)</span>
<span id="cb7-37"><a href="analysis.html#cb7-37" tabindex="-1"></a>    <span class="fu">do.call</span>(rbind, <span class="fu">lapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(unique_songs), <span class="cf">function</span>(i){</span>
<span id="cb7-38"><a href="analysis.html#cb7-38" tabindex="-1"></a>      sequence <span class="ot">&lt;-</span> shuffled_syls[<span class="fu">which</span>(data<span class="sc">$</span>song <span class="sc">==</span> unique_songs[i])]</span>
<span id="cb7-39"><a href="analysis.html#cb7-39" tabindex="-1"></a>      <span class="fu">return</span>(<span class="fu">matrix</span>(<span class="fu">c</span>(sequence[<span class="dv">1</span><span class="sc">:</span>(<span class="fu">length</span>(sequence)<span class="sc">-</span><span class="dv">1</span>)], sequence[<span class="dv">2</span><span class="sc">:</span><span class="fu">length</span>(sequence)]), <span class="at">ncol =</span> <span class="dv">2</span>))</span>
<span id="cb7-40"><a href="analysis.html#cb7-40" tabindex="-1"></a>    }))</span>
<span id="cb7-41"><a href="analysis.html#cb7-41" tabindex="-1"></a>  })</span>
<span id="cb7-42"><a href="analysis.html#cb7-42" tabindex="-1"></a></span>
<span id="cb7-43"><a href="analysis.html#cb7-43" tabindex="-1"></a>  <span class="co">#construct structured graph, ignoring edge weights</span></span>
<span id="cb7-44"><a href="analysis.html#cb7-44" tabindex="-1"></a>  graph <span class="ot">&lt;-</span> <span class="fu">simplify</span>(<span class="fu">graph_from_edgelist</span>(<span class="fu">unique</span>(directed_sequences), <span class="at">directed =</span> <span class="cn">TRUE</span>))</span>
<span id="cb7-45"><a href="analysis.html#cb7-45" tabindex="-1"></a></span>
<span id="cb7-46"><a href="analysis.html#cb7-46" tabindex="-1"></a>  <span class="co">#get small world indices for it</span></span>
<span id="cb7-47"><a href="analysis.html#cb7-47" tabindex="-1"></a>  structured_vals <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="dv">1</span><span class="sc">:</span>iters, <span class="cf">function</span>(x){</span>
<span id="cb7-48"><a href="analysis.html#cb7-48" tabindex="-1"></a>    temp <span class="ot">&lt;-</span> <span class="fu">sample_gnm</span>(<span class="at">n =</span> <span class="fu">length</span>(<span class="fu">V</span>(graph)), <span class="at">m =</span> <span class="fu">length</span>(<span class="fu">E</span>(graph)), <span class="at">directed =</span> <span class="cn">TRUE</span>)</span>
<span id="cb7-49"><a href="analysis.html#cb7-49" tabindex="-1"></a>    (<span class="fu">transitivity</span>(graph)<span class="sc">/</span><span class="fu">transitivity</span>(temp))<span class="sc">/</span>(<span class="fu">mean_distance</span>(graph)<span class="sc">/</span><span class="fu">mean_distance</span>(temp))</span>
<span id="cb7-50"><a href="analysis.html#cb7-50" tabindex="-1"></a>  })</span>
<span id="cb7-51"><a href="analysis.html#cb7-51" tabindex="-1"></a></span>
<span id="cb7-52"><a href="analysis.html#cb7-52" tabindex="-1"></a>  <span class="co">#get small world indices for null models</span></span>
<span id="cb7-53"><a href="analysis.html#cb7-53" tabindex="-1"></a>  null_vals <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="dv">1</span><span class="sc">:</span>iters, <span class="cf">function</span>(x){</span>
<span id="cb7-54"><a href="analysis.html#cb7-54" tabindex="-1"></a>    target_sequence <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="fu">as.numeric</span>(<span class="fu">factor</span>(shuffled_sequences[[x]])), <span class="at">ncol =</span> <span class="dv">2</span>)</span>
<span id="cb7-55"><a href="analysis.html#cb7-55" tabindex="-1"></a>    null <span class="ot">&lt;-</span> <span class="fu">simplify</span>(<span class="fu">graph_from_edgelist</span>(<span class="fu">unique</span>(target_sequence), <span class="at">directed =</span> <span class="cn">TRUE</span>))</span>
<span id="cb7-56"><a href="analysis.html#cb7-56" tabindex="-1"></a>    temp <span class="ot">&lt;-</span> <span class="fu">sample_gnm</span>(<span class="at">n =</span> <span class="fu">length</span>(<span class="fu">V</span>(null)), <span class="at">m =</span> <span class="fu">length</span>(<span class="fu">E</span>(null)), <span class="at">directed =</span> <span class="cn">TRUE</span>)</span>
<span id="cb7-57"><a href="analysis.html#cb7-57" tabindex="-1"></a>    (<span class="fu">transitivity</span>(null)<span class="sc">/</span><span class="fu">transitivity</span>(temp))<span class="sc">/</span>(<span class="fu">mean_distance</span>(null)<span class="sc">/</span><span class="fu">mean_distance</span>(temp))</span>
<span id="cb7-58"><a href="analysis.html#cb7-58" tabindex="-1"></a>  })</span>
<span id="cb7-59"><a href="analysis.html#cb7-59" tabindex="-1"></a></span>
<span id="cb7-60"><a href="analysis.html#cb7-60" tabindex="-1"></a>  <span class="co">#return objects</span></span>
<span id="cb7-61"><a href="analysis.html#cb7-61" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">list</span>(<span class="at">structured =</span> structured_vals, <span class="at">null =</span> null_vals))</span>
<span id="cb7-62"><a href="analysis.html#cb7-62" tabindex="-1"></a>}, <span class="at">mc.cores =</span> <span class="dv">5</span>)</span>
<span id="cb7-63"><a href="analysis.html#cb7-63" tabindex="-1"></a></span>
<span id="cb7-64"><a href="analysis.html#cb7-64" tabindex="-1"></a><span class="co">#save indices</span></span>
<span id="cb7-65"><a href="analysis.html#cb7-65" tabindex="-1"></a><span class="fu">save</span>(small_world, <span class="at">file =</span> <span class="st">&quot;data_models/small_world.RData&quot;</span>)</span></code></pre></div>
</details>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:figure-small-world"></span>
<img src="plots/small_world.png" alt="The left panel shows transition network between syllables types for a single male (#22 from 1975) above the global transition network for the entire dataset. The right panel shows the estimated small-worldness index calculated 1,000 times from the real data (blue) compared to pseudorandom sequences with the same frequency distribution of syllable types (orange). The dashed vertical line at 1 corresponds to the standard threshold for a network having small-world structure [@humphriesNetworkSmallworldnessQuantitative2008]." width="100%" />
<p class="caption">
Figure 4: The left panel shows transition network between syllables types for a single male (#22 from 1975) above the global transition network for the entire dataset. The right panel shows the estimated small-worldness index calculated 1,000 times from the real data (blue) compared to pseudorandom sequences with the same frequency distribution of syllable types (orange). The dashed vertical line at 1 corresponds to the standard threshold for a network having small-world structure <span class="citation">(<a href="#ref-humphriesNetworkSmallworldnessQuantitative2008">52</a>)</span>.
</p>
</div>
<p>Figure <a href="analysis.html#fig:figure-small-world">4</a>B shows the distribution of <span class="math inline">\(SWI\)</span> calculated from random sequences with the same distribution of types (orange) and the actual sequences (blue). At all three levels of granularity, the observed small-worldness of the real songs is above both the standard threshold (<span class="math inline">\(SWI &gt; 1\)</span>) and the pseudorandom sequences. <span class="math inline">\(SWI &gt; 1\)</span> when computed separately from the data from each year (1975, 2012, and 2019; see <a href="https://masonyoungblood.github.io/linguistic_efficiency/supplement/supplement.html">Supplementary Information</a>)</p>
</div>
<div id="mutual-information" class="section level2 hasAnchor" number="2.7">
<h2><span class="header-section-number">2.7</span> Mutual Information<a href="analysis.html#mutual-information" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The rate at which the mutual information between items decays with distance reflects whether underlying syntactic structure is Markovian, hierarchical, or a composite of the two <span class="citation">(<a href="#ref-sainburgParallelsSequentialOrganization2019">12</a>)</span>. Mutual information (<span class="math inline">\(MI\)</span>) was calculated from pairs of sequential syllables separated by a certain distance, using the method of <span class="citation">(<a href="#ref-sainburgParallelsSequentialOrganization2019">12</a>)</span>. <!--@sainburgParallelsSequentialOrganization2019 concatenated all songs recorded in a particular day before calculating $MI$. In our case, each day of recording represents several individuals recorded in several locations so concatenating sequences within days would lead to spurious pairs. Instead, --><span class="math inline">\(MI\)</span> was computed on concatenated sequences of songs from each individual. For example, if there are two individuals with two concatenated sequences of songs, <span class="math inline">\(a \to b \to c \to d\)</span> and <span class="math inline">\(e \to f \to g\)</span>, and the target distance is 2, then the pairs used for <span class="math inline">\(MI\)</span> calculated would be <span class="math inline">\(((a, c), (b, d), (e, g))\)</span>. <span class="math inline">\(MI\)</span> would then be calculated using:</p>
<p><span class="math display" id="eq:mut-info-a">\[\begin{equation}
\hat{I}(X, Y) = \hat{S}(X) + \hat{S}(Y) - \hat{S}(X, Y)
\tag{4}
\end{equation}\]</span></p>
<p>where <span class="math inline">\(\hat{I}\)</span> denotes information and <span class="math inline">\(\hat{S}\)</span> denotes entropy. <span class="math inline">\(X\)</span> is the distribution of first syllables <span class="math inline">\((a, b, e)\)</span>, <span class="math inline">\(Y\)</span> is the distribution of second syllables <span class="math inline">\((c, d, g)\)</span>, and <span class="math inline">\(XY\)</span> is the joint distribution of pairs <span class="math inline">\(((a, c), (b, d), (e, g))\)</span>. <span class="math inline">\(\hat{S}\)</span> was calculated using the method of <span class="citation">(<a href="#ref-grassberger08">96</a>)</span> and <span class="citation">(<a href="#ref-grassberger22">97</a>)</span> used by <span class="citation">(<a href="#ref-linCriticalBehaviorPhysics2017">67</a>)</span>:</p>
<p><span class="math display" id="eq:mut-info-b">\[\begin{equation}
\hat{S} = ln(N) - \frac{1}{N} \sum_{i=1}^{K} N_{i} \psi(N_{i})
\tag{5}
\end{equation}\]</span></p>
<p>where <span class="math inline">\(N\)</span> is the total number of tokens in the distribution, <span class="math inline">\(N_i\)</span> is the number of tokens within each type <span class="math inline">\(i\)</span> out of the total number of types <span class="math inline">\(K\)</span>, and <span class="math inline">\(\psi\)</span> is the digamma function. The actual <span class="math inline">\(MI\)</span> is then measured using:</p>
<p><span class="math display" id="eq:mut-info-c">\[\begin{equation}
MI = \hat{I} - \hat{I}_{sh}
\tag{6}
\end{equation}\]</span></p>
<p>where <span class="math inline">\(\hat{I}_{sh}\)</span> is the estimated lower bound of <span class="math inline">\(MI\)</span> calculated from shuffled sequences, created by randomly permuting the concatenated sequences of individuals’ songs.</p>
<p><span class="citation">(<a href="#ref-sainburgParallelsSequentialOrganization2019">12</a>)</span> and <span class="citation">(<a href="#ref-sainburg_etal22">68</a>)</span> simulated data from the hierarchical model of <span class="citation">(<a href="#ref-linCriticalBehaviorPhysics2017">67</a>)</span>, the Markov model of <span class="citation">(<a href="#ref-katahiraComplexSequencingRules2011a">98</a>)</span>, and their composite model in which Markov chains are nested within a larger hierarchical structure.</p>
<p>To determine whether the observed <span class="math inline">\(MI\)</span> decay was consistent with a Markov process, hierarchical process, or both, I fit the following three decay models to the data:</p>
<p><span class="math display" id="eq:mut-info-exp">\[\begin{equation}
\textrm{exponential decay: } y = ae^{-xb}
\tag{7}
\end{equation}\]</span>
<span class="math display" id="eq:mut-info-pl">\[\begin{equation}
\textrm{power-law decay: } y = cx^{d}
\tag{8}
\end{equation}\]</span>
<span class="math display" id="eq:mut-info-comp">\[\begin{equation}
\textrm{composite decay: } y = ae^{-xb} + cx^{d}
\tag{9}
\end{equation}\]</span></p>
<p>where <span class="math inline">\(y\)</span> is the estimated <span class="math inline">\(MI\)</span> and <span class="math inline">\(x\)</span> is the distance between syllables <span class="citation">(<a href="#ref-sainburgParallelsSequentialOrganization2019">12</a>,<a href="#ref-sainburg_etal22">68</a>)</span>. <span class="citation">(<a href="#ref-sainburgParallelsSequentialOrganization2019">12</a>)</span> included an intercept <span class="math inline">\(f\)</span> in all three models, but I removed it because it led to overfitting issues (e.g. exponential model with an extra “knee” after the initial decay), did not significantly improve model fit (<span class="math inline">\(\Delta WAIC &lt; 2\)</span>), and was not included in <span class="citation">(<a href="#ref-linCriticalBehaviorPhysics2017">67</a>)</span>. I focused my analysis on distances of up to 100 syllables to enable easy comparison with <span class="citation">(<a href="#ref-sainburgParallelsSequentialOrganization2019">12</a>)</span>. I followed <span class="citation">(<a href="#ref-sainburgParallelsSequentialOrganization2019">12</a>)</span> in comparing the fit of each model at increasing distances from 100 to 1,200 (the longest individual sequence is 1,219), and found that the composite model outperforms both the exponential and power-law models at all distances (see <a href="https://masonyoungblood.github.io/linguistic_efficiency/supplement/supplement.html">Supplementary Information</a>).</p>
<!-- Note that $MI$ decay becomes noisy and begins to slowly increase past ~400 syllables. This is because only a small number of individuals have concatenated sequences longer than that (~4%), and $MI$ is known to be systematically biased upwards at low sample sizes [@treves_panzeri95]. -->
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="analysis.html#cb8-1" tabindex="-1"></a><span class="co">#load packages</span></span>
<span id="cb8-2"><a href="analysis.html#cb8-2" tabindex="-1"></a><span class="fu">library</span>(brms)</span>
<span id="cb8-3"><a href="analysis.html#cb8-3" tabindex="-1"></a></span>
<span id="cb8-4"><a href="analysis.html#cb8-4" tabindex="-1"></a><span class="co">#load data</span></span>
<span id="cb8-5"><a href="analysis.html#cb8-5" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;data_models/processed_data.RData&quot;</span>)</span>
<span id="cb8-6"><a href="analysis.html#cb8-6" tabindex="-1"></a>data <span class="ot">&lt;-</span> processed_data<span class="sc">$</span>data</span>
<span id="cb8-7"><a href="analysis.html#cb8-7" tabindex="-1"></a>data<span class="sc">$</span>year <span class="ot">&lt;-</span> <span class="fu">strtrim</span>(data<span class="sc">$</span>individual, <span class="dv">4</span>)</span>
<span id="cb8-8"><a href="analysis.html#cb8-8" tabindex="-1"></a></span>
<span id="cb8-9"><a href="analysis.html#cb8-9" tabindex="-1"></a><span class="co">#create function for the grassberger method of entropy estimation used by lin and tegmark (log is ln by default, which is what grassberger used originally)</span></span>
<span id="cb8-10"><a href="analysis.html#cb8-10" tabindex="-1"></a>s_calc <span class="ot">&lt;-</span> <span class="cf">function</span>(vals){<span class="fu">return</span>(<span class="fu">log</span>(<span class="fu">sum</span>(vals)) <span class="sc">-</span> ((<span class="dv">1</span><span class="sc">/</span><span class="fu">sum</span>(vals))<span class="sc">*</span><span class="fu">sum</span>(vals<span class="sc">*</span><span class="fu">digamma</span>(vals))))}</span>
<span id="cb8-11"><a href="analysis.html#cb8-11" tabindex="-1"></a></span>
<span id="cb8-12"><a href="analysis.html#cb8-12" tabindex="-1"></a><span class="co">#get unique songs and song lengths</span></span>
<span id="cb8-13"><a href="analysis.html#cb8-13" tabindex="-1"></a>unique_songs <span class="ot">&lt;-</span> <span class="fu">unique</span>(data<span class="sc">$</span>song)</span>
<span id="cb8-14"><a href="analysis.html#cb8-14" tabindex="-1"></a>song_lengths <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">sapply</span>(unique_songs, <span class="cf">function</span>(x){<span class="fu">length</span>(<span class="fu">which</span>(data<span class="sc">$</span>song <span class="sc">==</span> x))}))</span>
<span id="cb8-15"><a href="analysis.html#cb8-15" tabindex="-1"></a></span>
<span id="cb8-16"><a href="analysis.html#cb8-16" tabindex="-1"></a><span class="co">#get unique individuals and the lengths of their sequences</span></span>
<span id="cb8-17"><a href="analysis.html#cb8-17" tabindex="-1"></a>unique_inds <span class="ot">&lt;-</span> <span class="fu">unique</span>(data<span class="sc">$</span>individual)</span>
<span id="cb8-18"><a href="analysis.html#cb8-18" tabindex="-1"></a>inds_lengths <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">sapply</span>(unique_inds, <span class="cf">function</span>(x){<span class="fu">length</span>(<span class="fu">which</span>(data<span class="sc">$</span>individual <span class="sc">==</span> x))}))</span>
<span id="cb8-19"><a href="analysis.html#cb8-19" tabindex="-1"></a></span>
<span id="cb8-20"><a href="analysis.html#cb8-20" tabindex="-1"></a><span class="co">#set random seed</span></span>
<span id="cb8-21"><a href="analysis.html#cb8-21" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">12345</span>)</span>
<span id="cb8-22"><a href="analysis.html#cb8-22" tabindex="-1"></a></span>
<span id="cb8-23"><a href="analysis.html#cb8-23" tabindex="-1"></a><span class="co">#calculate mutual information across three levels of deep split</span></span>
<span id="cb8-24"><a href="analysis.html#cb8-24" tabindex="-1"></a>mi_data <span class="ot">&lt;-</span> parallel<span class="sc">::</span><span class="fu">mclapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>, <span class="cf">function</span>(i){</span>
<span id="cb8-25"><a href="analysis.html#cb8-25" tabindex="-1"></a>  <span class="co">#use appropriate syllables</span></span>
<span id="cb8-26"><a href="analysis.html#cb8-26" tabindex="-1"></a>  <span class="cf">if</span>(i <span class="sc">==</span> <span class="dv">1</span>){data<span class="sc">$</span>syls <span class="ot">&lt;-</span> data<span class="sc">$</span>cluster_2}</span>
<span id="cb8-27"><a href="analysis.html#cb8-27" tabindex="-1"></a>  <span class="cf">if</span>(i <span class="sc">==</span> <span class="dv">2</span>){data<span class="sc">$</span>syls <span class="ot">&lt;-</span> data<span class="sc">$</span>cluster_3}</span>
<span id="cb8-28"><a href="analysis.html#cb8-28" tabindex="-1"></a>  <span class="cf">if</span>(i <span class="sc">==</span> <span class="dv">3</span>){data<span class="sc">$</span>syls <span class="ot">&lt;-</span> data<span class="sc">$</span>cluster_4}</span>
<span id="cb8-29"><a href="analysis.html#cb8-29" tabindex="-1"></a></span>
<span id="cb8-30"><a href="analysis.html#cb8-30" tabindex="-1"></a>  <span class="co">#compute actual mutual information curves</span></span>
<span id="cb8-31"><a href="analysis.html#cb8-31" tabindex="-1"></a>  actual <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="dv">1</span><span class="sc">:</span>(<span class="fu">max</span>(inds_lengths) <span class="sc">-</span> <span class="dv">1</span>), <span class="cf">function</span>(m){</span>
<span id="cb8-32"><a href="analysis.html#cb8-32" tabindex="-1"></a>    <span class="co">#set m as the distance between syllables in the sequence</span></span>
<span id="cb8-33"><a href="analysis.html#cb8-33" tabindex="-1"></a>    dist <span class="ot">&lt;-</span> m</span>
<span id="cb8-34"><a href="analysis.html#cb8-34" tabindex="-1"></a></span>
<span id="cb8-35"><a href="analysis.html#cb8-35" tabindex="-1"></a>    <span class="co">#store pairs within bouts</span></span>
<span id="cb8-36"><a href="analysis.html#cb8-36" tabindex="-1"></a>    pairs <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, <span class="fu">lapply</span>(unique_inds[<span class="fu">which</span>(inds_lengths <span class="sc">&gt;</span> m)], <span class="cf">function</span>(r){</span>
<span id="cb8-37"><a href="analysis.html#cb8-37" tabindex="-1"></a>      seq <span class="ot">&lt;-</span> data<span class="sc">$</span>syls[<span class="fu">which</span>(data<span class="sc">$</span>individual <span class="sc">==</span> r)]</span>
<span id="cb8-38"><a href="analysis.html#cb8-38" tabindex="-1"></a>      <span class="fu">t</span>(<span class="fu">sapply</span>(<span class="dv">1</span><span class="sc">:</span>(<span class="fu">length</span>(seq)<span class="sc">-</span>dist), <span class="cf">function</span>(x){<span class="fu">c</span>(seq[x], seq[x<span class="sc">+</span>dist])}))</span>
<span id="cb8-39"><a href="analysis.html#cb8-39" tabindex="-1"></a>    }))</span>
<span id="cb8-40"><a href="analysis.html#cb8-40" tabindex="-1"></a>    </span>
<span id="cb8-41"><a href="analysis.html#cb8-41" tabindex="-1"></a>    <span class="co">#get frequencies of syllables in each column</span></span>
<span id="cb8-42"><a href="analysis.html#cb8-42" tabindex="-1"></a>    n_x <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">table</span>(pairs[, <span class="dv">1</span>]))</span>
<span id="cb8-43"><a href="analysis.html#cb8-43" tabindex="-1"></a>    n_y <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">table</span>(pairs[, <span class="dv">2</span>]))</span>
<span id="cb8-44"><a href="analysis.html#cb8-44" tabindex="-1"></a></span>
<span id="cb8-45"><a href="analysis.html#cb8-45" tabindex="-1"></a>    <span class="co">#get frequencies of pairs in the data</span></span>
<span id="cb8-46"><a href="analysis.html#cb8-46" tabindex="-1"></a>    pairs <span class="ot">&lt;-</span> pairs[<span class="fu">order</span>(pairs[, <span class="dv">1</span>], pairs[, <span class="dv">2</span>]), ]</span>
<span id="cb8-47"><a href="analysis.html#cb8-47" tabindex="-1"></a>    index <span class="ot">&lt;-</span> <span class="sc">!</span><span class="fu">duplicated</span>(pairs)</span>
<span id="cb8-48"><a href="analysis.html#cb8-48" tabindex="-1"></a>    pair_freqs <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">tapply</span>(index, <span class="fu">cumsum</span>(index), length))</span>
<span id="cb8-49"><a href="analysis.html#cb8-49" tabindex="-1"></a></span>
<span id="cb8-50"><a href="analysis.html#cb8-50" tabindex="-1"></a>    <span class="co">#calculate s for each</span></span>
<span id="cb8-51"><a href="analysis.html#cb8-51" tabindex="-1"></a>    s_x <span class="ot">&lt;-</span> <span class="fu">s_calc</span>(n_x)</span>
<span id="cb8-52"><a href="analysis.html#cb8-52" tabindex="-1"></a>    s_y <span class="ot">&lt;-</span> <span class="fu">s_calc</span>(n_y)</span>
<span id="cb8-53"><a href="analysis.html#cb8-53" tabindex="-1"></a>    s_xy <span class="ot">&lt;-</span> <span class="fu">s_calc</span>(pair_freqs)</span>
<span id="cb8-54"><a href="analysis.html#cb8-54" tabindex="-1"></a></span>
<span id="cb8-55"><a href="analysis.html#cb8-55" tabindex="-1"></a>    <span class="co">#compute mutual information and return</span></span>
<span id="cb8-56"><a href="analysis.html#cb8-56" tabindex="-1"></a>    <span class="fu">return</span>(s_x <span class="sc">+</span> s_y <span class="sc">-</span> s_xy)</span>
<span id="cb8-57"><a href="analysis.html#cb8-57" tabindex="-1"></a>  })</span>
<span id="cb8-58"><a href="analysis.html#cb8-58" tabindex="-1"></a></span>
<span id="cb8-59"><a href="analysis.html#cb8-59" tabindex="-1"></a>  <span class="co">#compute shuffled mutual information curves</span></span>
<span id="cb8-60"><a href="analysis.html#cb8-60" tabindex="-1"></a>  shuffled <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="dv">1</span><span class="sc">:</span>(<span class="fu">max</span>(inds_lengths) <span class="sc">-</span> <span class="dv">1</span>), <span class="cf">function</span>(m){</span>
<span id="cb8-61"><a href="analysis.html#cb8-61" tabindex="-1"></a>    <span class="co">#set m as the distance between syllables in the sequence</span></span>
<span id="cb8-62"><a href="analysis.html#cb8-62" tabindex="-1"></a>    dist <span class="ot">&lt;-</span> m</span>
<span id="cb8-63"><a href="analysis.html#cb8-63" tabindex="-1"></a></span>
<span id="cb8-64"><a href="analysis.html#cb8-64" tabindex="-1"></a>    <span class="co">#store pairs within bouts</span></span>
<span id="cb8-65"><a href="analysis.html#cb8-65" tabindex="-1"></a>    pairs <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, <span class="fu">lapply</span>(unique_inds[<span class="fu">which</span>(inds_lengths <span class="sc">&gt;</span> m)], <span class="cf">function</span>(r){</span>
<span id="cb8-66"><a href="analysis.html#cb8-66" tabindex="-1"></a>      seq <span class="ot">&lt;-</span> <span class="fu">sample</span>(data<span class="sc">$</span>syls[<span class="fu">which</span>(data<span class="sc">$</span>individual <span class="sc">==</span> r)])</span>
<span id="cb8-67"><a href="analysis.html#cb8-67" tabindex="-1"></a>      <span class="fu">t</span>(<span class="fu">sapply</span>(<span class="dv">1</span><span class="sc">:</span>(<span class="fu">length</span>(seq)<span class="sc">-</span>dist), <span class="cf">function</span>(x){<span class="fu">c</span>(seq[x], seq[x<span class="sc">+</span>dist])}))</span>
<span id="cb8-68"><a href="analysis.html#cb8-68" tabindex="-1"></a>    }))</span>
<span id="cb8-69"><a href="analysis.html#cb8-69" tabindex="-1"></a></span>
<span id="cb8-70"><a href="analysis.html#cb8-70" tabindex="-1"></a>    <span class="co">#get frequencies of syllables in each column</span></span>
<span id="cb8-71"><a href="analysis.html#cb8-71" tabindex="-1"></a>    n_x <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">table</span>(pairs[, <span class="dv">1</span>]))</span>
<span id="cb8-72"><a href="analysis.html#cb8-72" tabindex="-1"></a>    n_y <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">table</span>(pairs[, <span class="dv">2</span>]))</span>
<span id="cb8-73"><a href="analysis.html#cb8-73" tabindex="-1"></a></span>
<span id="cb8-74"><a href="analysis.html#cb8-74" tabindex="-1"></a>    <span class="co">#get frequencies of pairs in the data</span></span>
<span id="cb8-75"><a href="analysis.html#cb8-75" tabindex="-1"></a>    pairs <span class="ot">&lt;-</span> pairs[<span class="fu">order</span>(pairs[, <span class="dv">1</span>], pairs[, <span class="dv">2</span>]), ]</span>
<span id="cb8-76"><a href="analysis.html#cb8-76" tabindex="-1"></a>    index <span class="ot">&lt;-</span> <span class="sc">!</span><span class="fu">duplicated</span>(pairs)</span>
<span id="cb8-77"><a href="analysis.html#cb8-77" tabindex="-1"></a>    pair_freqs <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">tapply</span>(index, <span class="fu">cumsum</span>(index), length))</span>
<span id="cb8-78"><a href="analysis.html#cb8-78" tabindex="-1"></a></span>
<span id="cb8-79"><a href="analysis.html#cb8-79" tabindex="-1"></a>    <span class="co">#calculate s for each</span></span>
<span id="cb8-80"><a href="analysis.html#cb8-80" tabindex="-1"></a>    s_x <span class="ot">&lt;-</span> <span class="fu">s_calc</span>(n_x)</span>
<span id="cb8-81"><a href="analysis.html#cb8-81" tabindex="-1"></a>    s_y <span class="ot">&lt;-</span> <span class="fu">s_calc</span>(n_y)</span>
<span id="cb8-82"><a href="analysis.html#cb8-82" tabindex="-1"></a>    s_xy <span class="ot">&lt;-</span> <span class="fu">s_calc</span>(pair_freqs)</span>
<span id="cb8-83"><a href="analysis.html#cb8-83" tabindex="-1"></a></span>
<span id="cb8-84"><a href="analysis.html#cb8-84" tabindex="-1"></a>    <span class="co">#compute mutual information and return</span></span>
<span id="cb8-85"><a href="analysis.html#cb8-85" tabindex="-1"></a>    <span class="fu">return</span>(s_x <span class="sc">+</span> s_y <span class="sc">-</span> s_xy)</span>
<span id="cb8-86"><a href="analysis.html#cb8-86" tabindex="-1"></a>  })</span>
<span id="cb8-87"><a href="analysis.html#cb8-87" tabindex="-1"></a></span>
<span id="cb8-88"><a href="analysis.html#cb8-88" tabindex="-1"></a>  <span class="co">#return mutual information decay, with shuffled baseline substracted from actual information</span></span>
<span id="cb8-89"><a href="analysis.html#cb8-89" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">data.frame</span>(<span class="at">dist =</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(actual), <span class="at">mi =</span> <span class="fu">c</span>(actual <span class="sc">-</span> shuffled), <span class="at">mi_raw =</span> actual, <span class="at">mi_base =</span> shuffled))</span>
<span id="cb8-90"><a href="analysis.html#cb8-90" tabindex="-1"></a>}, <span class="at">mc.cores =</span> <span class="dv">3</span>)</span>
<span id="cb8-91"><a href="analysis.html#cb8-91" tabindex="-1"></a></span>
<span id="cb8-92"><a href="analysis.html#cb8-92" tabindex="-1"></a><span class="co">#save output</span></span>
<span id="cb8-93"><a href="analysis.html#cb8-93" tabindex="-1"></a><span class="fu">save</span>(mi_data, <span class="at">file =</span> <span class="st">&quot;data_models/mi_data.RData&quot;</span>)</span></code></pre></div>
</details>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="analysis.html#cb9-1" tabindex="-1"></a><span class="co">#load packages</span></span>
<span id="cb9-2"><a href="analysis.html#cb9-2" tabindex="-1"></a><span class="fu">library</span>(brms)</span>
<span id="cb9-3"><a href="analysis.html#cb9-3" tabindex="-1"></a></span>
<span id="cb9-4"><a href="analysis.html#cb9-4" tabindex="-1"></a><span class="co">#load data</span></span>
<span id="cb9-5"><a href="analysis.html#cb9-5" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;data_models/mi_data.RData&quot;</span>)</span>
<span id="cb9-6"><a href="analysis.html#cb9-6" tabindex="-1"></a></span>
<span id="cb9-7"><a href="analysis.html#cb9-7" tabindex="-1"></a><span class="co">#set priors</span></span>
<span id="cb9-8"><a href="analysis.html#cb9-8" tabindex="-1"></a>exponential_priors <span class="ot">&lt;-</span> <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="dv">1</span>), <span class="at">lb =</span> <span class="dv">0</span>, <span class="at">nlpar =</span> <span class="st">&quot;a&quot;</span>) <span class="sc">+</span> </span>
<span id="cb9-9"><a href="analysis.html#cb9-9" tabindex="-1"></a>  <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="dv">1</span>), <span class="at">lb =</span> <span class="dv">0</span>, <span class="at">nlpar =</span> <span class="st">&quot;b&quot;</span>)</span>
<span id="cb9-10"><a href="analysis.html#cb9-10" tabindex="-1"></a>power_law_priors <span class="ot">&lt;-</span> <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="dv">1</span>), <span class="at">lb =</span> <span class="dv">0</span>, <span class="at">nlpar =</span> <span class="st">&quot;c&quot;</span>) <span class="sc">+</span> </span>
<span id="cb9-11"><a href="analysis.html#cb9-11" tabindex="-1"></a>  <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="dv">1</span>), <span class="at">lb =</span> <span class="dv">0</span>, <span class="at">nlpar =</span> <span class="st">&quot;d&quot;</span>)</span>
<span id="cb9-12"><a href="analysis.html#cb9-12" tabindex="-1"></a>composite_priors <span class="ot">&lt;-</span> <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="dv">1</span>), <span class="at">lb =</span> <span class="dv">0</span>, <span class="at">nlpar =</span> <span class="st">&quot;a&quot;</span>) <span class="sc">+</span> </span>
<span id="cb9-13"><a href="analysis.html#cb9-13" tabindex="-1"></a>  <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="dv">1</span>), <span class="at">lb =</span> <span class="dv">0</span>, <span class="at">nlpar =</span> <span class="st">&quot;b&quot;</span>) <span class="sc">+</span> </span>
<span id="cb9-14"><a href="analysis.html#cb9-14" tabindex="-1"></a>  <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="dv">1</span>), <span class="at">lb =</span> <span class="dv">0</span>, <span class="at">nlpar =</span> <span class="st">&quot;c&quot;</span>) <span class="sc">+</span> </span>
<span id="cb9-15"><a href="analysis.html#cb9-15" tabindex="-1"></a>  <span class="fu">prior</span>(<span class="fu">normal</span>(<span class="dv">0</span>, <span class="dv">1</span>), <span class="at">lb =</span> <span class="dv">0</span>, <span class="at">nlpar =</span> <span class="st">&quot;d&quot;</span>)</span>
<span id="cb9-16"><a href="analysis.html#cb9-16" tabindex="-1"></a></span>
<span id="cb9-17"><a href="analysis.html#cb9-17" tabindex="-1"></a><span class="co">#find the point at which the distances are no longer best fit by a composite model for each level of deep split</span></span>
<span id="cb9-18"><a href="analysis.html#cb9-18" tabindex="-1"></a>increments <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">100</span>, <span class="dv">200</span>, <span class="dv">300</span>, <span class="dv">400</span>, <span class="dv">500</span>, <span class="dv">600</span>, <span class="dv">700</span>, <span class="dv">800</span>, <span class="dv">900</span>, <span class="dv">1000</span>, <span class="dv">1200</span>)</span>
<span id="cb9-19"><a href="analysis.html#cb9-19" tabindex="-1"></a>waic_test <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>, <span class="cf">function</span>(h){</span>
<span id="cb9-20"><a href="analysis.html#cb9-20" tabindex="-1"></a>  parallel<span class="sc">::</span><span class="fu">mclapply</span>(increments, <span class="cf">function</span>(x){</span>
<span id="cb9-21"><a href="analysis.html#cb9-21" tabindex="-1"></a>    exponential_test <span class="ot">&lt;-</span> <span class="fu">brm</span>(<span class="fu">bf</span>(mi <span class="sc">~</span> a<span class="sc">*</span><span class="fu">exp</span>((<span class="sc">-</span>dist)<span class="sc">*</span>b), a <span class="sc">+</span> b <span class="sc">~</span> <span class="dv">1</span>, <span class="at">nl =</span> <span class="cn">TRUE</span>), </span>
<span id="cb9-22"><a href="analysis.html#cb9-22" tabindex="-1"></a>                            <span class="at">data =</span> mi_data[[h]][<span class="dv">1</span><span class="sc">:</span>x, ], <span class="at">prior =</span> exponential_priors, <span class="at">iter =</span> <span class="dv">5000</span>)</span>
<span id="cb9-23"><a href="analysis.html#cb9-23" tabindex="-1"></a>    power_law_test <span class="ot">&lt;-</span> <span class="fu">brm</span>(<span class="fu">bf</span>(mi <span class="sc">~</span> c<span class="sc">*</span>(dist<span class="sc">^</span>(<span class="sc">-</span>d)), c <span class="sc">+</span> d <span class="sc">~</span> <span class="dv">1</span>, <span class="at">nl =</span> <span class="cn">TRUE</span>), </span>
<span id="cb9-24"><a href="analysis.html#cb9-24" tabindex="-1"></a>                          <span class="at">data =</span> mi_data[[h]][<span class="dv">1</span><span class="sc">:</span>x, ], <span class="at">prior =</span> power_law_priors, <span class="at">iter =</span> <span class="dv">5000</span>)</span>
<span id="cb9-25"><a href="analysis.html#cb9-25" tabindex="-1"></a>    composite_test <span class="ot">&lt;-</span> <span class="fu">brm</span>(<span class="fu">bf</span>(mi <span class="sc">~</span> a<span class="sc">*</span><span class="fu">exp</span>((<span class="sc">-</span>dist)<span class="sc">*</span>b) <span class="sc">+</span> c<span class="sc">*</span>(dist<span class="sc">^</span>(<span class="sc">-</span>d)), a <span class="sc">+</span> b <span class="sc">+</span> c <span class="sc">+</span> d <span class="sc">~</span> <span class="dv">1</span>, <span class="at">nl =</span> <span class="cn">TRUE</span>), </span>
<span id="cb9-26"><a href="analysis.html#cb9-26" tabindex="-1"></a>                          <span class="at">data =</span> mi_data[[h]][<span class="dv">1</span><span class="sc">:</span>x, ], <span class="at">prior =</span> composite_priors, <span class="at">iter =</span> <span class="dv">5000</span>)</span>
<span id="cb9-27"><a href="analysis.html#cb9-27" tabindex="-1"></a>    waic_temp <span class="ot">&lt;-</span> <span class="fu">WAIC</span>(exponential_test, power_law_test, composite_test)</span>
<span id="cb9-28"><a href="analysis.html#cb9-28" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">c</span>(waic_temp<span class="sc">$</span>loos<span class="sc">$</span>exponential_test<span class="sc">$</span>waic, </span>
<span id="cb9-29"><a href="analysis.html#cb9-29" tabindex="-1"></a>             waic_temp<span class="sc">$</span>loos<span class="sc">$</span>power_law_test<span class="sc">$</span>waic,</span>
<span id="cb9-30"><a href="analysis.html#cb9-30" tabindex="-1"></a>             waic_temp<span class="sc">$</span>loos<span class="sc">$</span>composite_test<span class="sc">$</span>waic))</span>
<span id="cb9-31"><a href="analysis.html#cb9-31" tabindex="-1"></a>  }, <span class="at">mc.cores =</span> <span class="dv">7</span>)</span>
<span id="cb9-32"><a href="analysis.html#cb9-32" tabindex="-1"></a>})</span>
<span id="cb9-33"><a href="analysis.html#cb9-33" tabindex="-1"></a></span>
<span id="cb9-34"><a href="analysis.html#cb9-34" tabindex="-1"></a><span class="co">#format and save</span></span>
<span id="cb9-35"><a href="analysis.html#cb9-35" tabindex="-1"></a>waic_test <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>, <span class="cf">function</span>(x){</span>
<span id="cb9-36"><a href="analysis.html#cb9-36" tabindex="-1"></a>  temp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">do.call</span>(rbind, waic_test[[x]]))</span>
<span id="cb9-37"><a href="analysis.html#cb9-37" tabindex="-1"></a>  <span class="fu">colnames</span>(temp) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;exponential&quot;</span>, <span class="st">&quot;power_law&quot;</span>, <span class="st">&quot;composite&quot;</span>)</span>
<span id="cb9-38"><a href="analysis.html#cb9-38" tabindex="-1"></a>  <span class="fu">rownames</span>(temp) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">100</span>, <span class="dv">200</span>, <span class="dv">300</span>, <span class="dv">400</span>, <span class="dv">500</span>, <span class="dv">600</span>, <span class="dv">700</span>, <span class="dv">800</span>, <span class="dv">900</span>, <span class="dv">1000</span>, <span class="dv">1200</span>)</span>
<span id="cb9-39"><a href="analysis.html#cb9-39" tabindex="-1"></a>  <span class="fu">return</span>(temp)</span>
<span id="cb9-40"><a href="analysis.html#cb9-40" tabindex="-1"></a>})</span>
<span id="cb9-41"><a href="analysis.html#cb9-41" tabindex="-1"></a><span class="fu">save</span>(waic_test, <span class="at">file =</span> <span class="st">&quot;data_models/waic_test.RData&quot;</span>)</span>
<span id="cb9-42"><a href="analysis.html#cb9-42" tabindex="-1"></a></span>
<span id="cb9-43"><a href="analysis.html#cb9-43" tabindex="-1"></a><span class="co">#run models at all three levels of deep split</span></span>
<span id="cb9-44"><a href="analysis.html#cb9-44" tabindex="-1"></a>mi_models <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>, <span class="cf">function</span>(g){</span>
<span id="cb9-45"><a href="analysis.html#cb9-45" tabindex="-1"></a>  <span class="co">#fit the exponential model</span></span>
<span id="cb9-46"><a href="analysis.html#cb9-46" tabindex="-1"></a>  exponential_fit <span class="ot">&lt;-</span> <span class="fu">brm</span>(<span class="fu">bf</span>(mi <span class="sc">~</span> a<span class="sc">*</span><span class="fu">exp</span>((<span class="sc">-</span>dist)<span class="sc">*</span>b), a <span class="sc">+</span> b <span class="sc">~</span> <span class="dv">1</span>, <span class="at">nl =</span> <span class="cn">TRUE</span>),</span>
<span id="cb9-47"><a href="analysis.html#cb9-47" tabindex="-1"></a>                         <span class="at">data =</span> mi_data[[g]][<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>, ], <span class="at">prior =</span> exponential_priors,</span>
<span id="cb9-48"><a href="analysis.html#cb9-48" tabindex="-1"></a>                         <span class="at">iter =</span> <span class="dv">5000</span>, <span class="at">cores =</span> <span class="dv">4</span>)</span>
<span id="cb9-49"><a href="analysis.html#cb9-49" tabindex="-1"></a>  </span>
<span id="cb9-50"><a href="analysis.html#cb9-50" tabindex="-1"></a>  <span class="co">#fit the power law model</span></span>
<span id="cb9-51"><a href="analysis.html#cb9-51" tabindex="-1"></a>  power_law_fit <span class="ot">&lt;-</span> <span class="fu">brm</span>(<span class="fu">bf</span>(mi <span class="sc">~</span> c<span class="sc">*</span>(dist<span class="sc">^</span>(<span class="sc">-</span>d)), c <span class="sc">+</span> d <span class="sc">~</span> <span class="dv">1</span>, <span class="at">nl =</span> <span class="cn">TRUE</span>),</span>
<span id="cb9-52"><a href="analysis.html#cb9-52" tabindex="-1"></a>                       <span class="at">data =</span> mi_data[[g]][<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>, ], <span class="at">prior =</span> power_law_priors, </span>
<span id="cb9-53"><a href="analysis.html#cb9-53" tabindex="-1"></a>                       <span class="at">iter =</span> <span class="dv">5000</span>, <span class="at">cores =</span> <span class="dv">4</span>)</span>
<span id="cb9-54"><a href="analysis.html#cb9-54" tabindex="-1"></a>  </span>
<span id="cb9-55"><a href="analysis.html#cb9-55" tabindex="-1"></a>  <span class="co">#fit the composite model</span></span>
<span id="cb9-56"><a href="analysis.html#cb9-56" tabindex="-1"></a>  composite_fit <span class="ot">&lt;-</span> <span class="fu">brm</span>(<span class="fu">bf</span>(mi <span class="sc">~</span> a<span class="sc">*</span><span class="fu">exp</span>((<span class="sc">-</span>dist)<span class="sc">*</span>b) <span class="sc">+</span> c<span class="sc">*</span>(dist<span class="sc">^</span>(<span class="sc">-</span>d)), a <span class="sc">+</span> b <span class="sc">+</span> c <span class="sc">+</span> d <span class="sc">~</span> <span class="dv">1</span>, <span class="at">nl =</span> <span class="cn">TRUE</span>), </span>
<span id="cb9-57"><a href="analysis.html#cb9-57" tabindex="-1"></a>                       <span class="at">data =</span> mi_data[[g]][<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>, ], <span class="at">prior =</span> composite_priors, </span>
<span id="cb9-58"><a href="analysis.html#cb9-58" tabindex="-1"></a>                       <span class="at">iter =</span> <span class="dv">5000</span>, <span class="at">cores =</span> <span class="dv">4</span>)</span>
<span id="cb9-59"><a href="analysis.html#cb9-59" tabindex="-1"></a>  </span>
<span id="cb9-60"><a href="analysis.html#cb9-60" tabindex="-1"></a>  <span class="co">#calculate r2</span></span>
<span id="cb9-61"><a href="analysis.html#cb9-61" tabindex="-1"></a>  r2 <span class="ot">&lt;-</span> <span class="fu">rbind</span>(<span class="fu">bayes_R2</span>(exponential_fit), <span class="fu">bayes_R2</span>(power_law_fit), <span class="fu">bayes_R2</span>(composite_fit))</span>
<span id="cb9-62"><a href="analysis.html#cb9-62" tabindex="-1"></a>  <span class="fu">rownames</span>(r2) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;exponential&quot;</span>, <span class="st">&quot;power_law&quot;</span>, <span class="st">&quot;composite&quot;</span>)</span>
<span id="cb9-63"><a href="analysis.html#cb9-63" tabindex="-1"></a></span>
<span id="cb9-64"><a href="analysis.html#cb9-64" tabindex="-1"></a>  <span class="co">#compute aic</span></span>
<span id="cb9-65"><a href="analysis.html#cb9-65" tabindex="-1"></a>  waic <span class="ot">&lt;-</span> <span class="fu">WAIC</span>(exponential_fit, power_law_fit, composite_fit)</span>
<span id="cb9-66"><a href="analysis.html#cb9-66" tabindex="-1"></a>  waic <span class="ot">&lt;-</span> <span class="fu">c</span>(waic<span class="sc">$</span>loos<span class="sc">$</span>exponential_fit<span class="sc">$</span>estimates[<span class="dv">3</span>, <span class="dv">1</span>], waic<span class="sc">$</span>loos<span class="sc">$</span>power_law_fit<span class="sc">$</span>estimates[<span class="dv">3</span>, <span class="dv">1</span>], waic<span class="sc">$</span>loos<span class="sc">$</span>composite_fit<span class="sc">$</span>estimates[<span class="dv">3</span>, <span class="dv">1</span>])</span>
<span id="cb9-67"><a href="analysis.html#cb9-67" tabindex="-1"></a></span>
<span id="cb9-68"><a href="analysis.html#cb9-68" tabindex="-1"></a>  <span class="co">#return data and model fits and aic</span></span>
<span id="cb9-69"><a href="analysis.html#cb9-69" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">list</span>(<span class="at">waic =</span> waic,</span>
<span id="cb9-70"><a href="analysis.html#cb9-70" tabindex="-1"></a>              <span class="at">r2 =</span> r2,</span>
<span id="cb9-71"><a href="analysis.html#cb9-71" tabindex="-1"></a>              <span class="at">priors =</span> <span class="fu">prior_summary</span>(composite_fit),</span>
<span id="cb9-72"><a href="analysis.html#cb9-72" tabindex="-1"></a>              <span class="at">exponential_fit =</span> <span class="fu">summary</span>(exponential_fit)<span class="sc">$</span>fixed,</span>
<span id="cb9-73"><a href="analysis.html#cb9-73" tabindex="-1"></a>              <span class="at">power_law_fit =</span> <span class="fu">summary</span>(power_law_fit)<span class="sc">$</span>fixed,</span>
<span id="cb9-74"><a href="analysis.html#cb9-74" tabindex="-1"></a>              <span class="at">composite_fit =</span> <span class="fu">summary</span>(composite_fit)<span class="sc">$</span>fixed))</span>
<span id="cb9-75"><a href="analysis.html#cb9-75" tabindex="-1"></a>})</span>
<span id="cb9-76"><a href="analysis.html#cb9-76" tabindex="-1"></a></span>
<span id="cb9-77"><a href="analysis.html#cb9-77" tabindex="-1"></a><span class="co">#save data and models</span></span>
<span id="cb9-78"><a href="analysis.html#cb9-78" tabindex="-1"></a><span class="fu">save</span>(mi_models, <span class="at">file =</span> <span class="st">&quot;data_models/mi_models.RData&quot;</span>)</span></code></pre></div>
</details>
<details class=chunk-details><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="analysis.html#cb10-1" tabindex="-1"></a><span class="co">#extract simulated values from sainburg et al.</span></span>
<span id="cb10-2"><a href="analysis.html#cb10-2" tabindex="-1"></a>model_fig <span class="ot">&lt;-</span> xml2<span class="sc">::</span><span class="fu">read_xml</span>(<span class="st">&quot;https://raw.githubusercontent.com/timsainb/ParallelsBirdsongLanguagePaper/master/figures/modelfig.svg&quot;</span>)</span>
<span id="cb10-3"><a href="analysis.html#cb10-3" tabindex="-1"></a>xml2<span class="sc">::</span><span class="fu">xml_ns_strip</span>(model_fig)</span>
<span id="cb10-4"><a href="analysis.html#cb10-4" tabindex="-1"></a></span>
<span id="cb10-5"><a href="analysis.html#cb10-5" tabindex="-1"></a><span class="co">#convert raw values to actual values</span></span>
<span id="cb10-6"><a href="analysis.html#cb10-6" tabindex="-1"></a>raw_to_log <span class="ot">&lt;-</span> <span class="cf">function</span>(y, <span class="at">range =</span> <span class="fu">c</span>(<span class="fl">281.188594</span>, <span class="fl">6.838594</span>), <span class="at">axis =</span> <span class="fu">c</span>(<span class="fl">10e-4</span>, <span class="fl">10e1</span>)){<span class="fu">return</span>(<span class="fu">exp</span>(scales<span class="sc">::</span><span class="fu">rescale</span>(<span class="at">x =</span> <span class="sc">-</span><span class="fu">c</span>(range, y), <span class="at">from =</span> <span class="sc">-</span>range, <span class="at">to =</span> <span class="fu">log</span>(axis)))[<span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>)])}</span>
<span id="cb10-7"><a href="analysis.html#cb10-7" tabindex="-1"></a></span>
<span id="cb10-8"><a href="analysis.html#cb10-8" tabindex="-1"></a><span class="co">#get heirarchical values</span></span>
<span id="cb10-9"><a href="analysis.html#cb10-9" tabindex="-1"></a>y_heirarchy <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(xml2<span class="sc">::</span><span class="fu">xml_attr</span>(xml2<span class="sc">::</span><span class="fu">xml_find_all</span>(model_fig, <span class="at">xpath =</span> <span class="st">&quot;//*[@id=&#39;PathCollection_1&#39;]/g/use&quot;</span>), <span class="st">&quot;y&quot;</span>))</span>
<span id="cb10-10"><a href="analysis.html#cb10-10" tabindex="-1"></a>y_heirarchy <span class="ot">&lt;-</span> <span class="fu">raw_to_log</span>(y_heirarchy)</span>
<span id="cb10-11"><a href="analysis.html#cb10-11" tabindex="-1"></a></span>
<span id="cb10-12"><a href="analysis.html#cb10-12" tabindex="-1"></a><span class="co">#get markov values</span></span>
<span id="cb10-13"><a href="analysis.html#cb10-13" tabindex="-1"></a>y_markov_a <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(xml2<span class="sc">::</span><span class="fu">xml_attr</span>(xml2<span class="sc">::</span><span class="fu">xml_find_all</span>(model_fig, <span class="at">xpath =</span> <span class="st">&quot;//*[@id=&#39;PathCollection_2&#39;]/g/use&quot;</span>), <span class="st">&quot;y&quot;</span>))[<span class="dv">1</span><span class="sc">:</span><span class="dv">29</span>]</span>
<span id="cb10-14"><a href="analysis.html#cb10-14" tabindex="-1"></a>y_markov_b <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(xml2<span class="sc">::</span><span class="fu">xml_attr</span>(xml2<span class="sc">::</span><span class="fu">xml_find_all</span>(model_fig, <span class="at">xpath =</span> <span class="st">&quot;//*[@id=&#39;PathCollection_3&#39;]/g/use&quot;</span>), <span class="st">&quot;y&quot;</span>))[<span class="dv">1</span><span class="sc">:</span><span class="dv">12</span>]</span>
<span id="cb10-15"><a href="analysis.html#cb10-15" tabindex="-1"></a>y_markov_a <span class="ot">&lt;-</span> <span class="fu">raw_to_log</span>(y_markov_a)</span>
<span id="cb10-16"><a href="analysis.html#cb10-16" tabindex="-1"></a>y_markov_b <span class="ot">&lt;-</span> <span class="fu">raw_to_log</span>(y_markov_b)</span>
<span id="cb10-17"><a href="analysis.html#cb10-17" tabindex="-1"></a></span>
<span id="cb10-18"><a href="analysis.html#cb10-18" tabindex="-1"></a><span class="co">#get combined values</span></span>
<span id="cb10-19"><a href="analysis.html#cb10-19" tabindex="-1"></a>y_combined <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(xml2<span class="sc">::</span><span class="fu">xml_attr</span>(xml2<span class="sc">::</span><span class="fu">xml_find_all</span>(model_fig, <span class="at">xpath =</span> <span class="st">&quot;//*[@id=&#39;PathCollection_4&#39;]/g/use&quot;</span>), <span class="st">&quot;y&quot;</span>))</span>
<span id="cb10-20"><a href="analysis.html#cb10-20" tabindex="-1"></a>y_combined <span class="ot">&lt;-</span> <span class="fu">raw_to_log</span>(y_combined)</span>
<span id="cb10-21"><a href="analysis.html#cb10-21" tabindex="-1"></a></span>
<span id="cb10-22"><a href="analysis.html#cb10-22" tabindex="-1"></a><span class="co">#combine all simulated values into one data frame</span></span>
<span id="cb10-23"><a href="analysis.html#cb10-23" tabindex="-1"></a>mi_simulated <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">x =</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">12</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>), </span>
<span id="cb10-24"><a href="analysis.html#cb10-24" tabindex="-1"></a>                           <span class="at">y =</span> <span class="fu">c</span>(y_heirarchy, y_markov_b, y_combined),</span>
<span id="cb10-25"><a href="analysis.html#cb10-25" tabindex="-1"></a>                           <span class="at">group =</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">&quot;heirarchy&quot;</span>, <span class="dv">100</span>), <span class="fu">rep</span>(<span class="st">&quot;markov&quot;</span>, <span class="dv">12</span>), <span class="fu">rep</span>(<span class="st">&quot;combined&quot;</span>, <span class="dv">100</span>)))</span>
<span id="cb10-26"><a href="analysis.html#cb10-26" tabindex="-1"></a></span>
<span id="cb10-27"><a href="analysis.html#cb10-27" tabindex="-1"></a><span class="co">#save simulated values</span></span>
<span id="cb10-28"><a href="analysis.html#cb10-28" tabindex="-1"></a><span class="fu">save</span>(mi_simulated, <span class="at">file =</span> <span class="st">&quot;data_models/mi_simulated.RData&quot;</span>)</span></code></pre></div>
</details>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:figure-mutual-info"></span>
<img src="plots/mut_info.png" alt="The top panel shows simulated mutual information decay curves from the @linCriticalBehaviorPhysics2017 hierarchical model (left), the @katahiraComplexSequencingRules2011a Markov model (center), and the @sainburgParallelsSequentialOrganization2019 composite model (right) [data and inspiration for diagrams from @sainburgParallelsSequentialOrganization2019]. The bottom panel shows the computed mutual information decay curves for the observed data at different levels of deep split (left, center, right). The solid line corresponds to the full composite model, while the dashed and dotted lines correspond to the exponential and power-law terms, respectively." width="100%" />
<p class="caption">
Figure 5: The top panel shows simulated mutual information decay curves from the <span class="citation">(<a href="#ref-linCriticalBehaviorPhysics2017">67</a>)</span> hierarchical model (left), the <span class="citation">(<a href="#ref-katahiraComplexSequencingRules2011a">98</a>)</span> Markov model (center), and the <span class="citation">(<a href="#ref-sainburgParallelsSequentialOrganization2019">12</a>)</span> composite model (right) <span class="citation">(data and inspiration for diagrams from <a href="#ref-sainburgParallelsSequentialOrganization2019">12</a>)</span>. The bottom panel shows the computed mutual information decay curves for the observed data at different levels of deep split (left, center, right). The solid line corresponds to the full composite model, while the dashed and dotted lines correspond to the exponential and power-law terms, respectively.
</p>
</div>
<p>At all three levels of granularity, the composite model has a lower <span class="math inline">\(WAIC\)</span> and higher <span class="math inline">\(R^2\)</span> than both the exponential and power-law models (see <a href="https://masonyoungblood.github.io/linguistic_efficiency/supplement/supplement.html">Supplementary Information</a>), suggesting that mutual information decay in house finch song is more consistent with a combination of Markovian and hierarchical processes rather than with one or the other. The composite model also outcompetes both the exponential and power-law models when fit to mutual information decay curves computed separately for each year of recording (1975, 2012, and 2019; see <a href="https://masonyoungblood.github.io/linguistic_efficiency/supplement/supplement.html">Supplementary Information</a>).</p>
</div>
</div>
<h3>References</h3>
<div id="refs" class="references csl-bib-body">
<div id="ref-youngbloodContentBiasCultural2022" class="csl-entry">
<div class="csl-left-margin">9. </div><div class="csl-right-inline">Youngblood M, Lahti D. <a href="https://doi.org/10.1016/j.anbehav.2021.12.012">Content bias in the cultural evolution of house finch song</a>. Animal Behaviour. 2022;185:37–48. </div>
</div>
<div id="ref-sainburgParallelsSequentialOrganization2019" class="csl-entry">
<div class="csl-left-margin">12. </div><div class="csl-right-inline">Sainburg T, Theilman B, Thielk M, Gentner TQ. Parallels in the sequential organization of birdsong and human speech. Nat Commun [Internet]. 2019 Aug 12 [cited 2023 Apr 17];10(1):3636. Available from: <a href="https://www.nature.com/articles/s41467-019-11605-y">https://www.nature.com/articles/s41467-019-11605-y</a></div>
</div>
<div id="ref-raviv_etal21" class="csl-entry">
<div class="csl-left-margin">13. </div><div class="csl-right-inline">Raviv L, De Heer Kloots M, Meyer A. What makes a language easy to learn? A preregistered study on how systematic structure and community size affect language learnability. Cognition [Internet]. 2021 May [cited 2023 Aug 9];210:104620. Available from: <a href="https://linkinghub.elsevier.com/retrieve/pii/S0010027721000391">https://linkinghub.elsevier.com/retrieve/pii/S0010027721000391</a></div>
</div>
<div id="ref-Mandelbrot1953" class="csl-entry">
<div class="csl-left-margin">22. </div><div class="csl-right-inline">Mandelbrot B. An informational theory of the statistical structure of language. Communication Theory. 1953;486–502. </div>
</div>
<div id="ref-Mandelbrot1962" class="csl-entry">
<div class="csl-left-margin">23. </div><div class="csl-right-inline">Mandelbrot B. On the theory of word frequencies and on related Markovian models of discourse. 1962;190–219. </div>
</div>
<div id="ref-allenNetworkAnalysisReveals2019" class="csl-entry">
<div class="csl-left-margin">24. </div><div class="csl-right-inline">Allen JA, Garland EC, Dunlop RA, Noad MJ. Network analysis reveals underlying syntactic features in a vocally learnt mammalian display, humpback whale song. Proc R Soc B [Internet]. 2019 Dec 18 [cited 2023 May 29];286(1917):20192014. Available from: <a href="https://royalsocietypublishing.org/doi/10.1098/rspb.2019.2014">https://royalsocietypublishing.org/doi/10.1098/rspb.2019.2014</a></div>
</div>
<div id="ref-jamesPhylogenyMechanismsShared2021" class="csl-entry">
<div class="csl-left-margin">49. </div><div class="csl-right-inline">James LS, Mori C, Wada K, Sakata JT. Phylogeny and mechanisms of shared hierarchical patterns in birdsong. Current Biology [Internet]. 2021 Jul [cited 2023 Apr 12];31(13):2796–2808.e9. Available from: <a href="https://linkinghub.elsevier.com/retrieve/pii/S0960982221005285">https://linkinghub.elsevier.com/retrieve/pii/S0960982221005285</a></div>
</div>
<div id="ref-humphriesNetworkSmallworldnessQuantitative2008" class="csl-entry">
<div class="csl-left-margin">52. </div><div class="csl-right-inline">Humphries MD, Gurney K. Network <span>“small-world-ness”</span>: a quantitative method for determining canonical network equivalence. Sporns O, editor. PLoS ONE [Internet]. 2008 Apr 30 [cited 2023 May 29];3(4):e0002051. Available from: <a href="https://dx.plos.org/10.1371/journal.pone.0002051">https://dx.plos.org/10.1371/journal.pone.0002051</a></div>
</div>
<div id="ref-linCriticalBehaviorPhysics2017" class="csl-entry">
<div class="csl-left-margin">67. </div><div class="csl-right-inline">Lin H, Tegmark M. Critical behavior in physics and probabilistic formal languages. Entropy [Internet]. 2017 Jun 23 [cited 2023 May 29];19(7):299. Available from: <a href="http://www.mdpi.com/1099-4300/19/7/299">http://www.mdpi.com/1099-4300/19/7/299</a></div>
</div>
<div id="ref-sainburg_etal22" class="csl-entry">
<div class="csl-left-margin">68. </div><div class="csl-right-inline">Sainburg T, Mai A, Gentner TQ. Long-range sequential dependencies precede complex syntactic production in language acquisition. Proceedings of the Royal Society B: Biological Sciences [Internet]. 2022;289:20212657. Available from: <a href="https://doi.org/10.1098/rspb.2021.2657">https://doi.org/10.1098/rspb.2021.2657</a></div>
</div>
<div id="ref-Mundinger1975" class="csl-entry">
<div class="csl-left-margin">74. </div><div class="csl-right-inline">Mundinger PC. Song dialects and colonization in the house finch, Carpodacus mexicanus, on the east coast. The Condor [Internet]. 1975;77(4):407–22. Available from: <a href="https://doi.org/10.2307/1366088">https://doi.org/10.2307/1366088</a></div>
</div>
<div id="ref-Ju2019" class="csl-entry">
<div class="csl-left-margin">75. </div><div class="csl-right-inline">Ju C, Geller FC, Mundinger PC, Lahti DC. <a href="https://doi.org/10.1093/auk/uky012">Four decades of cultural evolution in house finch songs</a>. The Auk: Ornithological Advances. 2019;136:1–18. </div>
</div>
<div id="ref-burkner17" class="csl-entry">
<div class="csl-left-margin">81. </div><div class="csl-right-inline">Bürkner PC. brms: an R package for Bayesian multilevel models using Stan. J Stat Soft [Internet]. 2017 [cited 2023 Aug 15];80(1). Available from: <a href="http://www.jstatsoft.org/v80/i01/">http://www.jstatsoft.org/v80/i01/</a></div>
</div>
<div id="ref-watanabe21" class="csl-entry">
<div class="csl-left-margin">82. </div><div class="csl-right-inline">Watanabe S. Information criteria and cross validation for Bayesian inference in regular and singular cases. Jpn J Stat Data Sci [Internet]. 2021 Jul [cited 2023 Aug 15];4(1):1–19. Available from: <a href="https://link.springer.com/10.1007/s42081-021-00121-3">https://link.springer.com/10.1007/s42081-021-00121-3</a></div>
</div>
<div id="ref-gelman_etal19" class="csl-entry">
<div class="csl-left-margin">83. </div><div class="csl-right-inline">Gelman A, Goodrich B, Gabry J, Vehtari A. R-squared for Bayesian regression models. The American Statistician [Internet]. 2019 Jul 3 [cited 2023 Aug 15];73(3):307–9. Available from: <a href="https://www.tandfonline.com/doi/full/10.1080/00031305.2018.1549100">https://www.tandfonline.com/doi/full/10.1080/00031305.2018.1549100</a></div>
</div>
<div id="ref-Lachlan2018" class="csl-entry">
<div class="csl-left-margin">84. </div><div class="csl-right-inline">Lachlan RF, Ratmann O, Nowicki S. Cultural conformity generates extremely stable traditions in bird song. Nature Communications [Internet]. 2018;9. Available from: <a href="http://dx.doi.org/10.1038/s41467-018-04728-1">http://dx.doi.org/10.1038/s41467-018-04728-1</a></div>
</div>
<div id="ref-SardaEspinosa2019" class="csl-entry">
<div class="csl-left-margin">85. </div><div class="csl-right-inline">Sardá-Espinosa A. Time-series clustering in R using the dtwclust package. The R Journal [Internet]. 2019;11(1):22–43. Available from: <a href="https://doi.org/10.32614/RJ-2019-023">https://doi.org/10.32614/RJ-2019-023</a></div>
</div>
<div id="ref-Ratanamahatana2005" class="csl-entry">
<div class="csl-left-margin">86. </div><div class="csl-right-inline">Ratanamahatana CA, Keogh E. <a href="https://doi.org/10.1137/1.9781611972757.50">Three myths about dynamic time warping data mining</a>. In: Proceedings of the 2005 SIAM International Conference on Data Mining. 2005. p. 506–10. </div>
</div>
<div id="ref-Roginek2018" class="csl-entry">
<div class="csl-left-margin">87. </div><div class="csl-right-inline">Roginek EW. Spatial variation of house finch (Haemorhous mexicanus) song along the American Southwest coast. Queens College; 2018. </div>
</div>
<div id="ref-JSSv053i09" class="csl-entry">
<div class="csl-left-margin">88. </div><div class="csl-right-inline">Müllner D. fastcluster: Fast hierarchical, agglomerative clustering routines for R and python. Journal of Statistical Software [Internet]. 2013;53(9):1–18. Available from: <a href="https://www.jstatsoft.org/index.php/jss/article/view/v053i09">https://www.jstatsoft.org/index.php/jss/article/view/v053i09</a></div>
</div>
<div id="ref-langfelder_etal08" class="csl-entry">
<div class="csl-left-margin">89. </div><div class="csl-right-inline">Langfelder P, Zhang B, Horvath S. Defining clusters from a hierarchical cluster tree: the Dynamic Tree Cut package for R. Bioinformatics [Internet]. 2008 Mar 1 [cited 2023 Jun 2];24(5):719–20. Available from: <a href="https://academic.oup.com/bioinformatics/article/24/5/719/200751">https://academic.oup.com/bioinformatics/article/24/5/719/200751</a></div>
</div>
<div id="ref-podosFinescaleBroadlyApplicable2016" class="csl-entry">
<div class="csl-left-margin">90. </div><div class="csl-right-inline">Podos J, Moseley DL, Goodwin SE, McClure J, Taft BN, Strauss AVH, et al. <a href="https://doi.org/10.1016/j.anbehav.2016.03.036">A fine-scale, broadly applicable index of vocal performance: frequency excursion</a>. Animal Behaviour. 2016;116:203–12. </div>
</div>
<div id="ref-izsak06" class="csl-entry">
<div class="csl-left-margin">91. </div><div class="csl-right-inline">Izsák J. Some practical aspects of fitting and testing the Zipf-Mandelbrot model: a short essay. Scientometrics [Internet]. 2006 Apr [cited 2023 Sep 6];67(1):107–20. Available from: <a href="http://link.springer.com/10.1007/s11192-006-0052-x">http://link.springer.com/10.1007/s11192-006-0052-x</a></div>
</div>
<div id="ref-g_torre_etal21" class="csl-entry">
<div class="csl-left-margin">92. </div><div class="csl-right-inline">G. Torre I, Dębowski Ł, Hernández-Fernández A. Can Menzerath’s law be a criterion of complexity in communication? Amancio DR, editor. PLoS ONE [Internet]. 2021 Aug 20 [cited 2023 Aug 9];16(8):e0256133. Available from: <a href="https://dx.plos.org/10.1371/journal.pone.0256133">https://dx.plos.org/10.1371/journal.pone.0256133</a></div>
</div>
<div id="ref-ferrer-i-cancho_etal14" class="csl-entry">
<div class="csl-left-margin">94. </div><div class="csl-right-inline">Ferrer-i-Cancho R, Hernández-Fernández A, Baixeries J, Dębowski Ł, Mačutek J. When is Menzerath-Altmann law mathematically trivial? a new approach. Statistical Applications in Genetics and Molecular Biology [Internet]. 2014 Jan 1 [cited 2023 Aug 9];13(6). Available from: <a href="https://www.degruyter.com/document/doi/10.1515/sagmb-2013-0034/html">https://www.degruyter.com/document/doi/10.1515/sagmb-2013-0034/html</a></div>
</div>
<div id="ref-safryghinVariableExpressionLinguistic2022" class="csl-entry">
<div class="csl-left-margin">95. </div><div class="csl-right-inline">Safryghin A, Cross C, Fallon B, Heesen R, Ferrer-i-Cancho R, Hobaiter C. Variable expression of linguistic laws in ape gesture: a case study from chimpanzee sexual solicitation. Royal Society Open Science [Internet]. 2022;9:220849. Available from: <a href="https://doi.org/10.1098/rsos.220849">https://doi.org/10.1098/rsos.220849</a></div>
</div>
<div id="ref-grassberger08" class="csl-entry">
<div class="csl-left-margin">96. </div><div class="csl-right-inline">Grassberger P. Entropy estimates from insufficient samplings [Internet]. arXiv; 2008 [cited 2023 Aug 10]. Available from: <a href="http://arxiv.org/abs/physics/0307138">http://arxiv.org/abs/physics/0307138</a></div>
</div>
<div id="ref-grassberger22" class="csl-entry">
<div class="csl-left-margin">97. </div><div class="csl-right-inline">Grassberger P. On generalized Schürmann entropy estimators. Entropy [Internet]. 2022 May 11 [cited 2023 Aug 15];24(5):680. Available from: <a href="https://www.mdpi.com/1099-4300/24/5/680">https://www.mdpi.com/1099-4300/24/5/680</a></div>
</div>
<div id="ref-katahiraComplexSequencingRules2011a" class="csl-entry">
<div class="csl-left-margin">98. </div><div class="csl-right-inline">Katahira K, Suzuki K, Okanoya K, Okada M. Complex sequencing rules of birdsong can be explained by simple hidden Markov processes. De Polavieja GG, editor. PLoS ONE [Internet]. 2011 Sep 7 [cited 2023 May 29];6(9):e24516. Available from: <a href="https://dx.plos.org/10.1371/journal.pone.0024516">https://dx.plos.org/10.1371/journal.pone.0024516</a></div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="introduction.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="discussion.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": false,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
